{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from collections import defaultdict\n",
    "import json\n",
    "import pandas as pd\n",
    "from pathlib import Path\n",
    "import re\n",
    "import vaex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Data Paths\n",
    "root_path = Path.cwd().parents[0] \n",
    "data_dir = root_path / \"2020-03-13\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Load the metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "meta_df = pd.read_csv(data_dir / \"all_sources_metadata_2020-03-13.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sha</th>\n",
       "      <th>source_x</th>\n",
       "      <th>title</th>\n",
       "      <th>doi</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>license</th>\n",
       "      <th>abstract</th>\n",
       "      <th>publish_time</th>\n",
       "      <th>authors</th>\n",
       "      <th>journal</th>\n",
       "      <th>Microsoft Academic Paper ID</th>\n",
       "      <th>WHO #Covidence</th>\n",
       "      <th>has_full_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>c630ebcdf30652f0422c3ec12a00b50241dc9bd9</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Angiotensin-converting enzyme 2 (ACE2) as a SA...</td>\n",
       "      <td>10.1007/s00134-020-05985-9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32125455.0</td>\n",
       "      <td>cc-by-nc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Zhang, Haibo; Penninger, Josef M.; Li, Yimin; ...</td>\n",
       "      <td>Intensive Care Med</td>\n",
       "      <td>2.002765e+09</td>\n",
       "      <td>#3252</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>53eccda7977a31e3d0f565c884da036b1e85438e</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Comparative genetic analysis of the novel coro...</td>\n",
       "      <td>10.1038/s41421-020-0147-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Cao, Yanan; Li, Lin; Feng, Zhimin; Wan, Shengq...</td>\n",
       "      <td>Cell Discovery</td>\n",
       "      <td>3.003431e+09</td>\n",
       "      <td>#1861</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>210a892deb1c61577f6fba58505fd65356ce6636</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Incubation Period and Other Epidemiological Ch...</td>\n",
       "      <td>10.3390/jcm9020538</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>The geographic spread of 2019 novel coronaviru...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Linton, M. Natalie; Kobayashi, Tetsuro; Yang, ...</td>\n",
       "      <td>Journal of Clinical Medicine</td>\n",
       "      <td>3.006065e+09</td>\n",
       "      <td>#1043</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>e3b40cc8e0e137c416b4a2273a4dca94ae8178cc</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Characteristics of and Public Health Responses...</td>\n",
       "      <td>10.3390/jcm9020575</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32093211.0</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>In December 2019, cases of unidentified pneumo...</td>\n",
       "      <td>2020</td>\n",
       "      <td>Deng, Sheng-Qun; Peng, Hong-Juan</td>\n",
       "      <td>J Clin Med</td>\n",
       "      <td>1.776631e+08</td>\n",
       "      <td>#1999</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>92c2c9839304b4f2bc1276d41b1aa885d8b364fd</td>\n",
       "      <td>CZI</td>\n",
       "      <td>Imaging changes in severe COVID-19 pneumonia</td>\n",
       "      <td>10.1007/s00134-020-05976-w</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32125453.0</td>\n",
       "      <td>cc-by-nc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020</td>\n",
       "      <td>Zhang, Wei</td>\n",
       "      <td>Intensive Care Med</td>\n",
       "      <td>3.006643e+09</td>\n",
       "      <td>#3242</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        sha source_x  \\\n",
       "0  c630ebcdf30652f0422c3ec12a00b50241dc9bd9      CZI   \n",
       "1  53eccda7977a31e3d0f565c884da036b1e85438e      CZI   \n",
       "2  210a892deb1c61577f6fba58505fd65356ce6636      CZI   \n",
       "3  e3b40cc8e0e137c416b4a2273a4dca94ae8178cc      CZI   \n",
       "4  92c2c9839304b4f2bc1276d41b1aa885d8b364fd      CZI   \n",
       "\n",
       "                                               title  \\\n",
       "0  Angiotensin-converting enzyme 2 (ACE2) as a SA...   \n",
       "1  Comparative genetic analysis of the novel coro...   \n",
       "2  Incubation Period and Other Epidemiological Ch...   \n",
       "3  Characteristics of and Public Health Responses...   \n",
       "4       Imaging changes in severe COVID-19 pneumonia   \n",
       "\n",
       "                          doi pmcid   pubmed_id   license  \\\n",
       "0  10.1007/s00134-020-05985-9   NaN  32125455.0  cc-by-nc   \n",
       "1   10.1038/s41421-020-0147-1   NaN         NaN     cc-by   \n",
       "2          10.3390/jcm9020538   NaN         NaN     cc-by   \n",
       "3          10.3390/jcm9020575   NaN  32093211.0     cc-by   \n",
       "4  10.1007/s00134-020-05976-w   NaN  32125453.0  cc-by-nc   \n",
       "\n",
       "                                            abstract publish_time  \\\n",
       "0                                                NaN         2020   \n",
       "1                                                NaN         2020   \n",
       "2  The geographic spread of 2019 novel coronaviru...         2020   \n",
       "3  In December 2019, cases of unidentified pneumo...         2020   \n",
       "4                                                NaN         2020   \n",
       "\n",
       "                                             authors  \\\n",
       "0  Zhang, Haibo; Penninger, Josef M.; Li, Yimin; ...   \n",
       "1  Cao, Yanan; Li, Lin; Feng, Zhimin; Wan, Shengq...   \n",
       "2  Linton, M. Natalie; Kobayashi, Tetsuro; Yang, ...   \n",
       "3                   Deng, Sheng-Qun; Peng, Hong-Juan   \n",
       "4                                         Zhang, Wei   \n",
       "\n",
       "                        journal  Microsoft Academic Paper ID WHO #Covidence  \\\n",
       "0            Intensive Care Med                 2.002765e+09          #3252   \n",
       "1                Cell Discovery                 3.003431e+09          #1861   \n",
       "2  Journal of Clinical Medicine                 3.006065e+09          #1043   \n",
       "3                    J Clin Med                 1.776631e+08          #1999   \n",
       "4            Intensive Care Med                 3.006643e+09          #3242   \n",
       "\n",
       "  has_full_text  \n",
       "0          True  \n",
       "1          True  \n",
       "2          True  \n",
       "3          True  \n",
       "4         False  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 29500 entries, 0 to 29499\n",
      "Data columns (total 14 columns):\n",
      " #   Column                       Non-Null Count  Dtype  \n",
      "---  ------                       --------------  -----  \n",
      " 0   sha                          17420 non-null  object \n",
      " 1   source_x                     29500 non-null  object \n",
      " 2   title                        29130 non-null  object \n",
      " 3   doi                          26357 non-null  object \n",
      " 4   pmcid                        27337 non-null  object \n",
      " 5   pubmed_id                    16730 non-null  float64\n",
      " 6   license                      17692 non-null  object \n",
      " 7   abstract                     26553 non-null  object \n",
      " 8   publish_time                 18248 non-null  object \n",
      " 9   authors                      28554 non-null  object \n",
      " 10  journal                      17791 non-null  object \n",
      " 11  Microsoft Academic Paper ID  1134 non-null   float64\n",
      " 12  WHO #Covidence               1236 non-null   object \n",
      " 13  has_full_text                17420 non-null  object \n",
      "dtypes: float64(2), object(12)\n",
      "memory usage: 3.2+ MB\n"
     ]
    }
   ],
   "source": [
    "meta_df.info()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Not all papers have a title or abstract. Do all of them have at least one or the other?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(29130, 14)"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta_df[meta_df['title'].notnull() | meta_df['abstract'].notnull()].shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "No, looks like the limiting factor is the title"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sha</th>\n",
       "      <th>source_x</th>\n",
       "      <th>title</th>\n",
       "      <th>doi</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>license</th>\n",
       "      <th>abstract</th>\n",
       "      <th>publish_time</th>\n",
       "      <th>authors</th>\n",
       "      <th>journal</th>\n",
       "      <th>Microsoft Academic Paper ID</th>\n",
       "      <th>WHO #Covidence</th>\n",
       "      <th>has_full_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>18434</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC6824915</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23727</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC2762764</td>\n",
       "      <td>20025201.0</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2009 Sep 9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PLoS Curr</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23765</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC2762335</td>\n",
       "      <td>20020673.0</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2009 Aug 21</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PLoS Curr</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25385</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC2080411</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25598</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC1246105</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29495</th>\n",
       "      <td>c42a617a00afe6a36bde0a8e3638e0f55bfee4f7</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.08.20032847</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29496</th>\n",
       "      <td>c4ce14ce42fa4360dfe3515ec9d1584847381c27</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.08.20032854</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29497</th>\n",
       "      <td>c41e09a32be90c84cea0616bb1c726aecba721e0</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.09.20032219</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29498</th>\n",
       "      <td>ca88735399ff43d0e673876200655099f06f5567</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.09.20033183</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29499</th>\n",
       "      <td>0b48e1bfcdff9a42c88cc80a98661feb2703390d</td>\n",
       "      <td>medrxiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>doi.org/10.1101/2020.03.09.20033217</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>See https://www.medrxiv.org/submit-a-manuscript</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>370 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            sha source_x title  \\\n",
       "18434                                       NaN      PMC   NaN   \n",
       "23727                                       NaN      PMC   NaN   \n",
       "23765                                       NaN      PMC   NaN   \n",
       "25385                                       NaN      PMC   NaN   \n",
       "25598                                       NaN      PMC   NaN   \n",
       "...                                         ...      ...   ...   \n",
       "29495  c42a617a00afe6a36bde0a8e3638e0f55bfee4f7  medrxiv   NaN   \n",
       "29496  c4ce14ce42fa4360dfe3515ec9d1584847381c27  medrxiv   NaN   \n",
       "29497  c41e09a32be90c84cea0616bb1c726aecba721e0  medrxiv   NaN   \n",
       "29498  ca88735399ff43d0e673876200655099f06f5567  medrxiv   NaN   \n",
       "29499  0b48e1bfcdff9a42c88cc80a98661feb2703390d  medrxiv   NaN   \n",
       "\n",
       "                                       doi       pmcid   pubmed_id  \\\n",
       "18434                                  NaN  PMC6824915         NaN   \n",
       "23727                                  NaN  PMC2762764  20025201.0   \n",
       "23765                                  NaN  PMC2762335  20020673.0   \n",
       "25385                                  NaN  PMC2080411         NaN   \n",
       "25598                                  NaN  PMC1246105         NaN   \n",
       "...                                    ...         ...         ...   \n",
       "29495  doi.org/10.1101/2020.03.08.20032847         NaN         NaN   \n",
       "29496  doi.org/10.1101/2020.03.08.20032854         NaN         NaN   \n",
       "29497  doi.org/10.1101/2020.03.09.20032219         NaN         NaN   \n",
       "29498  doi.org/10.1101/2020.03.09.20033183         NaN         NaN   \n",
       "29499  doi.org/10.1101/2020.03.09.20033217         NaN         NaN   \n",
       "\n",
       "                                               license abstract publish_time  \\\n",
       "18434                                              NaN      NaN          NaN   \n",
       "23727                                            CC BY      NaN   2009 Sep 9   \n",
       "23765                                            CC BY      NaN  2009 Aug 21   \n",
       "25385                                              NaN      NaN          NaN   \n",
       "25598                                              NaN      NaN          NaN   \n",
       "...                                                ...      ...          ...   \n",
       "29495  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29496  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29497  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29498  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "29499  See https://www.medrxiv.org/submit-a-manuscript      NaN          NaN   \n",
       "\n",
       "      authors    journal  Microsoft Academic Paper ID WHO #Covidence  \\\n",
       "18434     NaN        NaN                          NaN            NaN   \n",
       "23727     NaN  PLoS Curr                          NaN            NaN   \n",
       "23765     NaN  PLoS Curr                          NaN            NaN   \n",
       "25385     NaN        NaN                          NaN            NaN   \n",
       "25598     NaN        NaN                          NaN            NaN   \n",
       "...       ...        ...                          ...            ...   \n",
       "29495     NaN        NaN                          NaN            NaN   \n",
       "29496     NaN        NaN                          NaN            NaN   \n",
       "29497     NaN        NaN                          NaN            NaN   \n",
       "29498     NaN        NaN                          NaN            NaN   \n",
       "29499     NaN        NaN                          NaN            NaN   \n",
       "\n",
       "      has_full_text  \n",
       "18434           NaN  \n",
       "23727           NaN  \n",
       "23765           NaN  \n",
       "25385           NaN  \n",
       "25598           NaN  \n",
       "...             ...  \n",
       "29495          True  \n",
       "29496          True  \n",
       "29497         False  \n",
       "29498          True  \n",
       "29499          True  \n",
       "\n",
       "[370 rows x 14 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta_df[meta_df['title'].isnull()]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Some of the papers without title seem to have full text. In our processing, we will filter out papers that have no textual fields, eg title, abstract, or full text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sha</th>\n",
       "      <th>source_x</th>\n",
       "      <th>title</th>\n",
       "      <th>doi</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>license</th>\n",
       "      <th>abstract</th>\n",
       "      <th>publish_time</th>\n",
       "      <th>authors</th>\n",
       "      <th>journal</th>\n",
       "      <th>Microsoft Academic Paper ID</th>\n",
       "      <th>WHO #Covidence</th>\n",
       "      <th>has_full_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>18434</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC6824915</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23727</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC2762764</td>\n",
       "      <td>20025201.0</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2009 Sep 9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PLoS Curr</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23765</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC2762335</td>\n",
       "      <td>20020673.0</td>\n",
       "      <td>CC BY</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2009 Aug 21</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PLoS Curr</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25385</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC2080411</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25598</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC1246105</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25648</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC1188130</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26186</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC404520</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26340</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC286345</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26629</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC1125910</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       sha source_x title  doi       pmcid   pubmed_id license abstract  \\\n",
       "18434  NaN      PMC   NaN  NaN  PMC6824915         NaN     NaN      NaN   \n",
       "23727  NaN      PMC   NaN  NaN  PMC2762764  20025201.0   CC BY      NaN   \n",
       "23765  NaN      PMC   NaN  NaN  PMC2762335  20020673.0   CC BY      NaN   \n",
       "25385  NaN      PMC   NaN  NaN  PMC2080411         NaN     NaN      NaN   \n",
       "25598  NaN      PMC   NaN  NaN  PMC1246105         NaN     NaN      NaN   \n",
       "25648  NaN      PMC   NaN  NaN  PMC1188130         NaN     NaN      NaN   \n",
       "26186  NaN      PMC   NaN  NaN   PMC404520         NaN     NaN      NaN   \n",
       "26340  NaN      PMC   NaN  NaN   PMC286345         NaN     NaN      NaN   \n",
       "26629  NaN      PMC   NaN  NaN  PMC1125910         NaN     NaN      NaN   \n",
       "\n",
       "      publish_time authors    journal  Microsoft Academic Paper ID  \\\n",
       "18434          NaN     NaN        NaN                          NaN   \n",
       "23727   2009 Sep 9     NaN  PLoS Curr                          NaN   \n",
       "23765  2009 Aug 21     NaN  PLoS Curr                          NaN   \n",
       "25385          NaN     NaN        NaN                          NaN   \n",
       "25598          NaN     NaN        NaN                          NaN   \n",
       "25648          NaN     NaN        NaN                          NaN   \n",
       "26186          NaN     NaN        NaN                          NaN   \n",
       "26340          NaN     NaN        NaN                          NaN   \n",
       "26629          NaN     NaN        NaN                          NaN   \n",
       "\n",
       "      WHO #Covidence has_full_text  \n",
       "18434            NaN           NaN  \n",
       "23727            NaN           NaN  \n",
       "23765            NaN           NaN  \n",
       "25385            NaN           NaN  \n",
       "25598            NaN           NaN  \n",
       "25648            NaN           NaN  \n",
       "26186            NaN           NaN  \n",
       "26340            NaN           NaN  \n",
       "26629            NaN           NaN  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta_df[meta_df['title'].isnull() & meta_df['abstract'].isnull() & meta_df['has_full_text'].isnull()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sha</th>\n",
       "      <th>title</th>\n",
       "      <th>doi</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>license</th>\n",
       "      <th>abstract</th>\n",
       "      <th>publish_time</th>\n",
       "      <th>authors</th>\n",
       "      <th>journal</th>\n",
       "      <th>Microsoft Academic Paper ID</th>\n",
       "      <th>WHO #Covidence</th>\n",
       "      <th>has_full_text</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>source_x</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CZI</th>\n",
       "      <td>167</td>\n",
       "      <td>1236</td>\n",
       "      <td>1177</td>\n",
       "      <td>0</td>\n",
       "      <td>404</td>\n",
       "      <td>172</td>\n",
       "      <td>848</td>\n",
       "      <td>1164</td>\n",
       "      <td>1191</td>\n",
       "      <td>1198</td>\n",
       "      <td>1134</td>\n",
       "      <td>1236</td>\n",
       "      <td>167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PMC</th>\n",
       "      <td>16326</td>\n",
       "      <td>27328</td>\n",
       "      <td>24253</td>\n",
       "      <td>27337</td>\n",
       "      <td>16326</td>\n",
       "      <td>16593</td>\n",
       "      <td>25139</td>\n",
       "      <td>16518</td>\n",
       "      <td>26797</td>\n",
       "      <td>16593</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>16326</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>biorxiv</th>\n",
       "      <td>566</td>\n",
       "      <td>566</td>\n",
       "      <td>566</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>566</td>\n",
       "      <td>566</td>\n",
       "      <td>566</td>\n",
       "      <td>566</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>566</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>medrxiv</th>\n",
       "      <td>361</td>\n",
       "      <td>0</td>\n",
       "      <td>361</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>361</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>361</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            sha  title    doi  pmcid  pubmed_id  license  abstract  \\\n",
       "source_x                                                             \n",
       "CZI         167   1236   1177      0        404      172       848   \n",
       "PMC       16326  27328  24253  27337      16326    16593     25139   \n",
       "biorxiv     566    566    566      0          0      566       566   \n",
       "medrxiv     361      0    361      0          0      361         0   \n",
       "\n",
       "          publish_time  authors  journal  Microsoft Academic Paper ID  \\\n",
       "source_x                                                                \n",
       "CZI               1164     1191     1198                         1134   \n",
       "PMC              16518    26797    16593                            0   \n",
       "biorxiv            566      566        0                            0   \n",
       "medrxiv              0        0        0                            0   \n",
       "\n",
       "          WHO #Covidence  has_full_text  \n",
       "source_x                                 \n",
       "CZI                 1236            167  \n",
       "PMC                    0          16326  \n",
       "biorxiv                0            566  \n",
       "medrxiv                0            361  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta_df.groupby('source_x').count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_jsons = list(data_dir.glob('**/*.json'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "13202"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(all_jsons)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "There is metadata for 29500 entries, but 13202 json files."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "PosixPath('/Users/akdidier/Documents/CORD/2020-03-13/pmc_custom_license/pmc_custom_license/8f8eb4f004c2002face0723f2f58cc411954d36e.json')"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_jsons[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_sets = [\"biorxiv_medrxiv\", \"comm_use_subset\", \"noncomm_use_subset\", \"pmc_custom_license\"]\n",
    "jsons = []\n",
    "for d in data_sets:\n",
    "    jsons.extend((data_dir / d).glob(\"**/*.json\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "PosixPath('/Users/akdidier/Documents/CORD/2020-03-13/biorxiv_medrxiv/biorxiv_medrxiv/f905f78b32f63c6d14a79984dfb33f1b358b8ab4.json')"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "jsons[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "13202"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(jsons)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(jsons[0], \"r\") as f:\n",
    "    data = json.load(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'paper_id': 'f905f78b32f63c6d14a79984dfb33f1b358b8ab4',\n",
       " 'metadata': {'title': 'Multimerization of HIV-1 integrase hinges on conserved SH3-docking platforms',\n",
       "  'authors': [{'first': 'Meytal',\n",
       "    'middle': [],\n",
       "    'last': 'Galilee',\n",
       "    'suffix': '',\n",
       "    'affiliation': {'laboratory': '',\n",
       "     'institution': 'Technion -Israel Institute of Technology',\n",
       "     'location': {'postCode': '320003, 320003',\n",
       "      'settlement': 'Haifa, Haifa',\n",
       "      'country': 'Israel., Israel'}},\n",
       "    'email': ''},\n",
       "   {'first': 'Akram',\n",
       "    'middle': [],\n",
       "    'last': 'Alian',\n",
       "    'suffix': '',\n",
       "    'affiliation': {'laboratory': '',\n",
       "     'institution': 'Technion -Israel Institute of Technology',\n",
       "     'location': {'postCode': '320003, 320003',\n",
       "      'settlement': 'Haifa, Haifa',\n",
       "      'country': 'Israel., Israel'}},\n",
       "    'email': 'alian@technion.ac.il'}]},\n",
       " 'abstract': [{'text': 'New anti-AIDS treatments must be continually developed in order to overcome resistance mutations including those emerging in the newest therapeutic target, the viral integrase (IN). Multimerization of IN is functionally imperative and provides a forthcoming therapeutic target. Allosteric inhibitors of IN bind to non-catalytic sites and prevent correct multimerization not only restricting viral integration but also the assembly and maturation of viral particles. Here, we report an allosteric inhibitor peptide targeting an unexploited SH3-docking platform of retroviral IN. The crystal structure of the peptide in complex with the HIV-1 IN core domain reveals a steric interference that would inhibit conserved docking of SH3-containing domain with the core domain vital for IN multimerization, providing a template for the development of novel anti-IN allosteric inhibitors.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Abstract'}],\n",
       " 'body_text': [{'text': \"In the absence of a curative treatment, the highly active antiretroviral therapy (HAART) keeps the HIV-1 virus of AIDS patients under control. HAART combines drugs targeting different stages of viral replication including the integration step catalyzed by the integrase protein (IN) (1) . Integration of viral DNA into host genome involves two steps catalyzed by IN: (i) cleavage of a dinucleotide from each 3'-end of the viral DNA (3'processing), and (ii) insertion of this processed viral DNA into the host DNA (strand-transfer) (2) . Clinical IN strand transfer inhibitors (INSTIs) target the catalytic site of the enzyme to specifically inhibit the DNA joining reaction, however, as with all anti-AIDS treatments, the continued success of these drugs is persistently disrupted by resistance mutations (1, 2) . Although 3'-processing can be carried out by monomeric IN (3) , the assembly of IN functional multimers is imperative for the strand-transfer activity (4) (5) (6) (7) (8) , and for virus particle maturation and production (reviewed in (9, 10) ). In the continued quest to identify and develop new drugs, allosteric inhibitors that bind sites outside the catalytic core and disrupt IN multimerization are emerging with potent therapeutic potential (11) (12) (13) (14) .\",\n",
       "   'cite_spans': [{'start': 283,\n",
       "     'end': 286,\n",
       "     'text': '(1)',\n",
       "     'ref_id': 'BIBREF0'},\n",
       "    {'start': 531, 'end': 534, 'text': '(2)', 'ref_id': 'BIBREF1'},\n",
       "    {'start': 805, 'end': 808, 'text': '(1,', 'ref_id': 'BIBREF0'},\n",
       "    {'start': 809, 'end': 811, 'text': '2)', 'ref_id': 'BIBREF1'},\n",
       "    {'start': 869, 'end': 875, 'text': 'IN (3)', 'ref_id': 'BIBREF2'},\n",
       "    {'start': 965, 'end': 968, 'text': '(4)', 'ref_id': 'BIBREF3'},\n",
       "    {'start': 969, 'end': 972, 'text': '(5)', 'ref_id': 'BIBREF4'},\n",
       "    {'start': 973, 'end': 976, 'text': '(6)', 'ref_id': 'BIBREF5'},\n",
       "    {'start': 977, 'end': 980, 'text': '(7)', 'ref_id': 'BIBREF6'},\n",
       "    {'start': 981, 'end': 984, 'text': '(8)', 'ref_id': 'BIBREF7'},\n",
       "    {'start': 1049, 'end': 1052, 'text': '(9,', 'ref_id': 'BIBREF8'},\n",
       "    {'start': 1053, 'end': 1056, 'text': '10)', 'ref_id': 'BIBREF9'},\n",
       "    {'start': 1261, 'end': 1265, 'text': '(11)', 'ref_id': 'BIBREF10'},\n",
       "    {'start': 1266, 'end': 1270, 'text': '(12)', 'ref_id': 'BIBREF11'},\n",
       "    {'start': 1271, 'end': 1275, 'text': '(13)', 'ref_id': 'BIBREF12'},\n",
       "    {'start': 1276, 'end': 1280, 'text': '(14)', 'ref_id': 'BIBREF13'}],\n",
       "   'ref_spans': [],\n",
       "   'section': ''},\n",
       "  {'text': 'Tetramers of IN are formed by the reciprocal swapping of the three, N-terminal, catalytic core and Cterminal, canonical domains of IN. The two internal IN protomers, where catalytic binding of viral and host DNA takes place, make up the majority of the tetramer interface. The outer two protomers, and other surrounding units in higher order assemblies, provide supportive domains indispensible for the assembly of the tetrameric cores (4) (5) (6) (7) (8) . We recently showed that targeting the interface between the N-terminal domain (NTD) and catalytic core domain (CCD) of IN, using a specific antigen-binding fragment (Fab), inhibits IN tetramer formation and consequent enzymatic activity and virus particle production (9) . Disrupting the dimerization of the elementary dimeric blocks building IN tetramers has also been explored as a strategy to inhibit IN activity (15) .',\n",
       "   'cite_spans': [{'start': 436,\n",
       "     'end': 439,\n",
       "     'text': '(4)',\n",
       "     'ref_id': 'BIBREF3'},\n",
       "    {'start': 440, 'end': 443, 'text': '(5)', 'ref_id': 'BIBREF4'},\n",
       "    {'start': 444, 'end': 447, 'text': '(6)', 'ref_id': 'BIBREF5'},\n",
       "    {'start': 448, 'end': 451, 'text': '(7)', 'ref_id': 'BIBREF6'},\n",
       "    {'start': 452, 'end': 455, 'text': '(8)', 'ref_id': 'BIBREF7'},\n",
       "    {'start': 725, 'end': 728, 'text': '(9)', 'ref_id': 'BIBREF8'},\n",
       "    {'start': 874, 'end': 878, 'text': '(15)', 'ref_id': 'BIBREF14'}],\n",
       "   'ref_spans': [],\n",
       "   'section': ''},\n",
       "  {'text': 'Hindering the assembly of IN functional multimers is only one side of the coin. Allosteric interference has also been shown to promote the formation of aberrant IN multimers and aggregates. The potential of allosteric IN inhibitors has been demonstrated through the thorough characterization of the \"LEDGF pocket\" formed at the dimer interface of IN and the development of LEDGIN (or ALLINI) inhibitors that bind to it (11) (Figure 1A) . Although less investigated, other IN pockets capable of allosteric inhibitor binding have also been identified ( Figure 1A) : binding of the \"Y3\" molecule to a pocket near the N-terminal end of CCD α-helix 4, designated Y3pocket, has been shown to inhibit 3\\'-processing and strand-transfer activities (16) ; the \"sucrose\" binding pocket found along the CCD dimer interface and flanked by two LEDGF pockets (17, 18) has recently been targeted by the natural product kuwanon-L, which inhibited IN activity in a pattern similar to LEDGINs (19) . Another fragment-binding pocket \"FBP\" has also been identified at the CCD dimer interface (20) . Although inhibitory profiles for this FBP pocket have not been shown, our previous work on IN from the feline virus showed that a single Phe187Lys mutation (Phe185 in HIV-1 IN) around this site inhibits dimer formation (3) . Fragment-based screening and structural studies have revealed the capability of all three pockets (Y3, FBP and LEDGF) to bind numerous small molecules (21) . Most recently, a new class of LEDGINs has been shown to distinctively bind to undefined interfaces of both CCD and CTD of IN (22) . Therefore, IN harbors several allosteric hotspots that can be explored for novel anti-multimerization intervention strategies.',\n",
       "   'cite_spans': [{'start': 419,\n",
       "     'end': 423,\n",
       "     'text': '(11)',\n",
       "     'ref_id': 'BIBREF10'},\n",
       "    {'start': 739, 'end': 743, 'text': '(16)', 'ref_id': 'BIBREF15'},\n",
       "    {'start': 844, 'end': 848, 'text': '(17,', 'ref_id': 'BIBREF16'},\n",
       "    {'start': 849, 'end': 852, 'text': '18)', 'ref_id': 'BIBREF17'},\n",
       "    {'start': 974, 'end': 978, 'text': '(19)', 'ref_id': 'BIBREF18'},\n",
       "    {'start': 1071, 'end': 1075, 'text': '(20)', 'ref_id': 'BIBREF19'},\n",
       "    {'start': 1297, 'end': 1300, 'text': '(3)', 'ref_id': 'BIBREF2'},\n",
       "    {'start': 1454, 'end': 1458, 'text': '(21)', 'ref_id': 'BIBREF20'},\n",
       "    {'start': 1586, 'end': 1590, 'text': '(22)', 'ref_id': 'BIBREF21'}],\n",
       "   'ref_spans': [{'start': 424,\n",
       "     'end': 435,\n",
       "     'text': '(Figure 1A)',\n",
       "     'ref_id': None},\n",
       "    {'start': 551, 'end': 561, 'text': 'Figure 1A)', 'ref_id': None}],\n",
       "   'section': ''},\n",
       "  {'text': 'Based on the structure of Fab specific to IN CCD/NTD platform (9), we develop a peptide that, similar to Fab, inhibits IN multimerization and strand-transfer activity. Surprisingly, crystal structure and affinity experiments show that the peptide interferes with the CCD/CTD interfaces of IN. This structure, which features an overlooked SH3-docking platform crucial for IN multimerization, can now provide a template for the screening and development of novel anti-IN allosteric inhibitors.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': ''},\n",
       "  {'text': \"Within a functional IN tetramer, the NTD and CTD of one dimer wrap around an extended α-helix (α4) of CCD of a second dimer ( Figure 1B) , an interdomaindocking platform that is functionally imperative (9) . Previously, we showed that complementarity-defining regions 3 (CDR3) and 2 (CDR2) of an anti-IN Fab wrap around α4 of IN CCD (residues 153-168) and dock into a cleft rimmed by the CCD finger loop (residues 186-191) ( Figure  1C ). We derived a peptide (WSYFYDGSYSYYDYESY) mostly representing the CDR3 sequence (underlined) with the addition of Trp (for UV-Vis absorbance) and Ser-Tyr (representing CDR2 extension in wrapping around α4). Similar to parental Fab, the peptide inhibited IN strand-transfer activity but not 3'processing ( Figure 1D ), suggesting potential interference with IN functional multimerization.\",\n",
       "   'cite_spans': [{'start': 202,\n",
       "     'end': 205,\n",
       "     'text': '(9)',\n",
       "     'ref_id': 'BIBREF8'}],\n",
       "   'ref_spans': [{'start': 126,\n",
       "     'end': 136,\n",
       "     'text': 'Figure 1B)',\n",
       "     'ref_id': None},\n",
       "    {'start': 425, 'end': 435, 'text': 'Figure  1C', 'ref_id': None},\n",
       "    {'start': 743, 'end': 752, 'text': 'Figure 1D', 'ref_id': None}],\n",
       "   'section': 'Fab derived peptide inhibits IN multimerization and strand-transfer activity'},\n",
       "  {'text': 'To evaluate the effects of the peptide on IN multimerization we used chemical crosslinking and found the peptide to indeed interfere with IN multimerization ( Figure 1E ). This underscores the allosteric potentials of the site(s) targeted by the peptide, and clarifies that the inhibition of IN multimerization and activity by the parental Fab (9) was specific to CDR3/CDR2 local allosteric interference and not an artifact of the bulkier size of whole Fab.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 159,\n",
       "     'end': 168,\n",
       "     'text': 'Figure 1E',\n",
       "     'ref_id': None}],\n",
       "   'section': 'Fab derived peptide inhibits IN multimerization and strand-transfer activity'},\n",
       "  {'text': 'To characterize the peptide binding site(s), we solved the 2.0 Å crystal structure of HIV-1 IN CCD in complex with the peptide (table-1). Except for the Cterminal pair of residues (Ser-Tyr), well-defined electron densities show binding of the peptide molecule to two sites, which we call site-1 and site-2 ( Figure 2A ). By examining available structures of various IN functional multimers (4) (5) (6) (7) (8) we underscore the conservation of these two CCD/CTD docking platforms ( Figure 2B ). Site-1 is at the tetrameic interface and is formed by a CTD from a flanking protomer docking at the carboxyl-terminus of CCD α4 of an inner protomer ( Figure 2B & 1B) . Forming site-2 is the docking of a CTD of another flanking protomer into the amino-terminus of the same α4 ( Figure  2B & 2C). The surface area buried upon peptide binding at site-2 is ~ 100 Å 2 larger than that at site-1 (744 as compared to 650 Å 2 ). Peptide binding at site-2 also involves a higher number of interacting residues at the interface (26 as compared to 17 at site-1). Side chains of two residues of the peptide (Trp1 and Tyr3 at site-1 and Ser10 and Tyr14 at site-2) participate in direct hydrogen bonds with the CCD (to side chains of Glu69 and Asp167 at site-1, and to carbonyl oxygen of Val79 and Ala80 at site-2). Additional hydrogen bonds were directly formed with the backbone of the peptide (one in site-1 and three in site-2), and water molecules mediated additional interactions at each site (two in site-1 and one in site-2) ( Figure 2D ). Interestingly, Ala80 carbonyl oxygen of CCD at site-2, which makes a hydrogen bond with the peptide, also makes a hydrogen bond (2.93 Å) with Lys266 of CTD in the HIV-1 IN functional multimers (discussed below and Figure 4B ). It has been shown that the positive charge at this Lys266 position is crucial for viral replication in which a Lys266Ala or Lys266Glu produced replication defective viruses (23) .',\n",
       "   'cite_spans': [{'start': 390,\n",
       "     'end': 393,\n",
       "     'text': '(4)',\n",
       "     'ref_id': 'BIBREF3'},\n",
       "    {'start': 394, 'end': 397, 'text': '(5)', 'ref_id': 'BIBREF4'},\n",
       "    {'start': 398, 'end': 401, 'text': '(6)', 'ref_id': 'BIBREF5'},\n",
       "    {'start': 402, 'end': 405, 'text': '(7)', 'ref_id': 'BIBREF6'},\n",
       "    {'start': 406, 'end': 409, 'text': '(8)', 'ref_id': 'BIBREF7'},\n",
       "    {'start': 1930, 'end': 1934, 'text': '(23)', 'ref_id': 'BIBREF22'}],\n",
       "   'ref_spans': [{'start': 308,\n",
       "     'end': 317,\n",
       "     'text': 'Figure 2A',\n",
       "     'ref_id': None},\n",
       "    {'start': 482, 'end': 491, 'text': 'Figure 2B', 'ref_id': None},\n",
       "    {'start': 646, 'end': 661, 'text': 'Figure 2B & 1B)', 'ref_id': None},\n",
       "    {'start': 773, 'end': 783, 'text': 'Figure  2B', 'ref_id': None},\n",
       "    {'start': 1517, 'end': 1526, 'text': 'Figure 2D', 'ref_id': None},\n",
       "    {'start': 1744, 'end': 1753, 'text': 'Figure 4B', 'ref_id': 'FIGREF2'}],\n",
       "   'section': 'The peptide targets two SH3-docking sites at the CCD/CTD interfaces'},\n",
       "  {'text': 'Whereas the peptide mainly represented the CDR3sequence and was expected to also represent CDR3 binding pattern (to dock at the CCD/NTD interface between α4 and the finger loop, (Figure 1C )), the peptide surprisingly docked underneath α4 to mostly resemble the binding of CDR2 at site-1 ( Figure 2A ). Superimposing the CCD-peptide structure onto that of the HIV-1 intasome reveals potential steric interference with the CCD docking platform of CTD but not NTD at site-1 ( Figure 2E , top). Similarly, the symmetry related molecule of the peptide bound at site-2 reveals potential interference with the docking of another CTD ( Figure 2E , bottom).',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 178,\n",
       "     'end': 188,\n",
       "     'text': '(Figure 1C',\n",
       "     'ref_id': None},\n",
       "    {'start': 290, 'end': 299, 'text': 'Figure 2A', 'ref_id': None},\n",
       "    {'start': 474, 'end': 483, 'text': 'Figure 2E', 'ref_id': None},\n",
       "    {'start': 629, 'end': 638, 'text': 'Figure 2E', 'ref_id': None}],\n",
       "   'section': 'The peptide targets two SH3-docking sites at the CCD/CTD interfaces'},\n",
       "  {'text': 'To demonstrate the ability of a fragment specific to site-2 to interfere with CCD/CTD docking and consequently inhibit IN multimerization, we evaluated the inhibitory effects of the Y3-molecule, which specifically binds at site-2 (16) . Indeed, using chemical crosslinking we show that the Y3-molecule attenuates IN . CC-BY 4.0 International license author/funder. It is made available under a The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/301721 doi: bioRxiv preprint multimerization in a dose-dependent manner ( Figure 2F ), corroborating on the potentials of a site-2 bound peptide to inhibit IN multimerizaion. Therefore, peptide binding at either of these sites, or both, would inhibit CTD interactions with CCD, and can rationalize the observed inhibition of IN multimerization ( Figure 1E ) and strand-transfer activity ( Figure 1D ). It is worth noting that since both docking sites involve not only the CTD but also the NTD linker segment (residues 50-58) ( Figure 2B ), peptide binding would also interfere with the vital docking of NTDs during IN multimerization.',\n",
       "   'cite_spans': [{'start': 230,\n",
       "     'end': 234,\n",
       "     'text': '(16)',\n",
       "     'ref_id': 'BIBREF15'}],\n",
       "   'ref_spans': [{'start': 570,\n",
       "     'end': 579,\n",
       "     'text': 'Figure 2F',\n",
       "     'ref_id': None},\n",
       "    {'start': 842, 'end': 851, 'text': 'Figure 1E', 'ref_id': None},\n",
       "    {'start': 885, 'end': 894, 'text': 'Figure 1D', 'ref_id': None},\n",
       "    {'start': 1023, 'end': 1032, 'text': 'Figure 2B', 'ref_id': None}],\n",
       "   'section': 'The peptide targets two SH3-docking sites at the CCD/CTD interfaces'},\n",
       "  {'text': \"Competency of peptide-bound IN for 3'processing activity indicates that peptide binding did not restrict the catalytic loop flexibility (density map is missing for residues 140-147) nor induced inactivating conformational changes to the enzyme (root-men-square deviation [RMSD] 0.30 Å to apo, PDB code: 1BIS).\",\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'The peptide targets two SH3-docking sites at the CCD/CTD interfaces'},\n",
       "  {'text': 'The higher number of interactions and the larger surface area buried upon peptide binding to site-2 may highlight a binding privilege at this site. Analyses of crystal structures of IN truncation variants, which have been shown to dimerize (CCD and CCDCTD) or tetramerize (NTDCCD) in solution and crystals, show the clear exposure and accessibility of site-1 ( Figure 3A and B). Whereas site-2 is also exposed in the crystal structures of dimeric CCD and tetrameric NTDCCD ( Figure 3A ), analyses of CCDCTD crystal structures intriguingly show that CTD is constantly docked into site-2, albeit in various configurations distinct to that of the functional intasome ( Figure 3B , orange cartoon). If indeed the CTD preoccupies site-2 within the truncated CCDCTD variant, as seen in crystal structures, then we expect that, unlike CCD or NTDCCD variants, the CCDCTD construct would not bind the peptide. Assessing peptide-binding affinities to the various IN constructs show that the presence of NTD (within NTDCCD) only slightly (~ 1.5 folds) interferes with peptide binding ( Figure 3C ). CTD (within CCDCTD), on the other hand, almost completely abolishes peptide binding to more than 12 folds ( Figure  3C ). In agreement with the binding data, structural superimposition of the peptide shows that while NTD (of NTDCCD structure) may delicately interfere with peptide binding at site-1 ( Figure 3A ), CTD (of CCDCTD structures) would provide a more robust interference at site-2 ( Figure 3B ). It is worth noting that the peptide bound to full-length IN, which is preassembled into tetramers in solution (24) , with affinities (Kd 1700 ± 88 nM) corresponding to the concentrations in the monomerization assay ( Figure 1E ) and ~ 4 folds weaker than NTDCCD (Kd 441 ± 11 nM, Figure 3C ). Together, the NTD appears to crucially modulate CTD interactions with the CCD.',\n",
       "   'cite_spans': [{'start': 1605,\n",
       "     'end': 1609,\n",
       "     'text': '(24)',\n",
       "     'ref_id': 'BIBREF23'}],\n",
       "   'ref_spans': [{'start': 361,\n",
       "     'end': 370,\n",
       "     'text': 'Figure 3A',\n",
       "     'ref_id': 'FIGREF0'},\n",
       "    {'start': 475, 'end': 484, 'text': 'Figure 3A', 'ref_id': 'FIGREF0'},\n",
       "    {'start': 666, 'end': 675, 'text': 'Figure 3B', 'ref_id': 'FIGREF0'},\n",
       "    {'start': 1075, 'end': 1084, 'text': 'Figure 3C', 'ref_id': 'FIGREF0'},\n",
       "    {'start': 1196, 'end': 1206, 'text': 'Figure  3C', 'ref_id': 'FIGREF0'},\n",
       "    {'start': 1389, 'end': 1398, 'text': 'Figure 3A', 'ref_id': 'FIGREF0'},\n",
       "    {'start': 1482, 'end': 1491, 'text': 'Figure 3B', 'ref_id': 'FIGREF0'},\n",
       "    {'start': 1712, 'end': 1721, 'text': 'Figure 1E', 'ref_id': None},\n",
       "    {'start': 1774, 'end': 1783, 'text': 'Figure 3C', 'ref_id': 'FIGREF0'}],\n",
       "   'section': 'SH3-docking platform-2 offers an ideal spot for allosteric targeting'},\n",
       "  {'text': \"The CTD preoccupation of site-2 in the twodomain CCDCTD construct may have significant implications on both the physiologic and the aberrant interactions mediated by the SH3-containing CTD domain. The incompetence of CCDCTD variant for 3'-processing activity (25, 26) indicates that the CCD/CTD configuration seen in the crystal structures of this construct ( Figure 3B ) is aberrant. Conceivably, interdomain interactions, especially those provided by the NTD and missing in CCDCTD, are required for correct CTD interactions and to maintain the catalytic site of the enzyme accessible for catalytic activities.\",\n",
       "   'cite_spans': [{'start': 259,\n",
       "     'end': 263,\n",
       "     'text': '(25,',\n",
       "     'ref_id': 'BIBREF24'},\n",
       "    {'start': 264, 'end': 267, 'text': '26)', 'ref_id': 'BIBREF25'}],\n",
       "   'ref_spans': [{'start': 360,\n",
       "     'end': 369,\n",
       "     'text': 'Figure 3B',\n",
       "     'ref_id': 'FIGREF0'}],\n",
       "   'section': 'Implications for the CTD interdomain interactions'},\n",
       "  {'text': \"To gain further insights into the interactions made by NTD and CTD, and their interference with the catalytic activity of CCD, we assessed 3'-processing activity of truncated domain variants of HIV-1 IN. These truncation variants are not active in strand-transfer activity (24) (25) (26) , a result that we also validated here (not shown). We found that whereas the CCD (residues 56-209) and NTDCCD (residues 1-209) variants of HIV-1 IN are as competent for the 3'-processing activity as the full-length enzyme, the CCDCTD (residues 56-288) is indeed defective ( Figure  4A ).\",\n",
       "   'cite_spans': [{'start': 273,\n",
       "     'end': 277,\n",
       "     'text': '(24)',\n",
       "     'ref_id': 'BIBREF23'},\n",
       "    {'start': 278, 'end': 282, 'text': '(25)', 'ref_id': 'BIBREF24'},\n",
       "    {'start': 283, 'end': 287, 'text': '(26)', 'ref_id': 'BIBREF25'}],\n",
       "   'ref_spans': [{'start': 563,\n",
       "     'end': 573,\n",
       "     'text': 'Figure  4A',\n",
       "     'ref_id': 'FIGREF2'}],\n",
       "   'section': 'Implications for the CTD interdomain interactions'},\n",
       "  {'text': \"The inhibitory effect of CTD, as part of the CCDCTD, was diminished upon separating the CTD segment (residues 210-288) and supplementing it to the CCD protein in the reaction mixture ( Figure 4A , CCD/CTD). Furthermore, the addition of CTD to a reaction containing the separate CCD and NTD (residues 1-55), which by itself ( Figure 4A , CCD/NTD) was found more than 3 folds superior to full-length or conjugate NTDCCD, resulted in a significant 33% inhibition ( Figure  4A , CCD/NTD/CTD). Therefore, while NTD positively interferes with the CCD in the 3'-processing activity, the CTD appears to have a negative effect. The contribution of the NTD appears necessary to support appropriate CTD interactions with the CCD. Structural analysis shows that the NTD linker segment (residues 50-58) contributes 55% (725 Å 2 ) of the total 1310 Å 2 buried CCD/CTD area at site-2 ( Figure 4B) . Interestingly, the carbonyl oxygen of Val54 of NTD linker forms a hydrogen bond (2.91 Å) with the side chain of CTD Arg228 (NH2) (Figure 4B ), mutating of which to Arg228Ala resulted in defective HIV-1 viruses (23) . The side chain of NTD Gln53 (NE2) forms a hydrogen bond (2.50 Å) with the carbonyl oxygen of CTD Asp229 (Figure 4B) , and also interacts with the side chain of Lys264 (3.6 Å). Both Lys264 and Lys266, which hydrogen bond to Ala80, have been shown to play an important role in the LEDGINs induced aberrant multimerization (27) . Indeed, the fundamental role of CTD . CC-BY 4.0 International license author/funder. It is made available under a The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/301721 doi: bioRxiv preprint in aberrant IN aggregation has previously been emphasized (10, 27, 28) . CTD interactions with the NTD linker (residues 44-53) at site-1, which provides the main docking template for CTD at this site, buries a total of 944 Å 2 ( Figure 4C) . Therefore, the NTD linker appears to play a crucial role in modulating the CCD/CTD docking platform and functional IN mutlimerization and is worth exploring in future studies.\",\n",
       "   'cite_spans': [{'start': 1094,\n",
       "     'end': 1098,\n",
       "     'text': '(23)',\n",
       "     'ref_id': 'BIBREF22'},\n",
       "    {'start': 1421, 'end': 1425, 'text': '(27)', 'ref_id': 'BIBREF26'},\n",
       "    {'start': 1731, 'end': 1735, 'text': '(10,', 'ref_id': 'BIBREF9'},\n",
       "    {'start': 1736, 'end': 1739, 'text': '27,', 'ref_id': 'BIBREF26'},\n",
       "    {'start': 1740, 'end': 1743, 'text': '28)', 'ref_id': 'BIBREF27'}],\n",
       "   'ref_spans': [{'start': 185,\n",
       "     'end': 194,\n",
       "     'text': 'Figure 4A',\n",
       "     'ref_id': 'FIGREF2'},\n",
       "    {'start': 325, 'end': 334, 'text': 'Figure 4A', 'ref_id': 'FIGREF2'},\n",
       "    {'start': 462, 'end': 472, 'text': 'Figure  4A', 'ref_id': 'FIGREF2'},\n",
       "    {'start': 871, 'end': 881, 'text': 'Figure 4B)', 'ref_id': 'FIGREF2'},\n",
       "    {'start': 1013, 'end': 1023, 'text': '(Figure 4B', 'ref_id': 'FIGREF2'},\n",
       "    {'start': 1205, 'end': 1216, 'text': '(Figure 4B)', 'ref_id': 'FIGREF2'},\n",
       "    {'start': 1902, 'end': 1912, 'text': 'Figure 4C)', 'ref_id': 'FIGREF2'}],\n",
       "   'section': 'Implications for the CTD interdomain interactions'},\n",
       "  {'text': 'Multimerization of HIV-1 IN depends on the delicate compatibility of amino acids lining the interacting interfaces (9), providing an attractive targeting strategy. Interfering with the interacting domains can be exploited not only to promote aberrant multimerization but also to inhibit the assembly of functional multimers, an opportunity offered by the CCD/NTD (HTH-cleft) (9) and the CCD/CTD (SH3-cleft) docking platform reported here. The SH3-cleft (site-2) relates to a previously identified \"Y3\" pocket observed almost two decades ago in a crystal structure of retroviral IN CCD bound to the small molecule (Y3) ( Figure 1A) , which efficiently inhibits IN catalytic activity (16) . A recent fragment-based screening study has also captured several other small molecules bound near this site (21) (Figure 1A, CDQ) , however, the inhibitory profiles where not characterized.',\n",
       "   'cite_spans': [{'start': 682,\n",
       "     'end': 686,\n",
       "     'text': '(16)',\n",
       "     'ref_id': 'BIBREF15'}],\n",
       "   'ref_spans': [{'start': 620,\n",
       "     'end': 630,\n",
       "     'text': 'Figure 1A)',\n",
       "     'ref_id': None},\n",
       "    {'start': 803, 'end': 819, 'text': '(Figure 1A, CDQ)', 'ref_id': None}],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'The inability of peptide to bind CCDCTD, which possesses an exposed site-1 but blocked site-2, indicates that site-2, which docks the CTD during multimerization, is a preferred binding site. CTD involvement in the multimerization of IN is well documented (reviewed in (23)), and the relative orientation of the CTD with regard to the CCD has previously been implicated in functional IN assembly (29) . Moreover, analysis of available CCD structures bound to small molecules at site-2 reveals a pocket that forms upon binding, which displaces the flexible N-terminus of α4, a displacement that would also interfere with and inhibit the binding of viral DNA and IN catalytic activity ( Figure 4D ). Therefore, we propose site-2 for future structure based design or screening of small molecules that can interfere with CCD/CTD interactions during IN multimerization, and can promote the formation of a binding pocket that would also inactivate the catalytic site by displacing α4 into an aberrant configuration.',\n",
       "   'cite_spans': [{'start': 395,\n",
       "     'end': 399,\n",
       "     'text': '(29)',\n",
       "     'ref_id': 'BIBREF28'}],\n",
       "   'ref_spans': [{'start': 684,\n",
       "     'end': 693,\n",
       "     'text': 'Figure 4D',\n",
       "     'ref_id': 'FIGREF2'}],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'The fact that IN regulates not only viral integration but also the assembly and maturation of virus particles (9, 22) , accentuates the prominence of this target. Whereas the peptide used in this study may not, per se, provide a lead hit for this anti-IN strategy, our work underscores the overlooked SH3-docking platform of HIV-1 IN as a potential therapeutic target for future anti-IN allosteric inhibitors.',\n",
       "   'cite_spans': [{'start': 110,\n",
       "     'end': 113,\n",
       "     'text': '(9,',\n",
       "     'ref_id': 'BIBREF8'},\n",
       "    {'start': 114, 'end': 117, 'text': '22)', 'ref_id': 'BIBREF21'}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'DISCUSSION'},\n",
       "  {'text': 'All HIV-1 IN constructs were derived from the previously described variant (SF1: (SF1; C56S, W131D, F185K and C280S (24)) and subcloned into pET28b plasmid (Novagen) with a cleavable N-terminal His-tag. The various constructs contained the following residues: full-length 1-288, NTD 1-55, CCD 56-209, NTDCCD 1-209, CTD 210-288 and CCDCTD 56-288. To include the NTD linker, the NTD 1-55 used here was made longer than our previous construct (residues 1-51 of HFH chimera (9)). All constructs were confirmed by DNA sequencing.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'EXPERIMENTAL PROCEDURES Preparation of HIV-1 IN Constructs:'},\n",
       "  {'text': 'IN constructs were expressed and purified as previously described (24) . Lysis buffer contained 50 mM potassium phosphate (pH 7.5), 5 mM Imidazole, 5 mM βmercaptoethanol and 1M NaCl for the full-length protein or 0. ',\n",
       "   'cite_spans': [{'start': 66,\n",
       "     'end': 70,\n",
       "     'text': '(24)',\n",
       "     'ref_id': 'BIBREF23'}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Protein Expression and Purification:'},\n",
       "  {'text': 'Fluorescence based IN activity assays were performed as previously described (30) . Fab was used at a 1:1 molar ratio and the peptide at an excessive ratio of ~ 20 folds. and were processed with iMosflm (31) . The structure was solved by Phaser molecular replacement (32) using IN CCD (PDB code: 1BIS (33)) as a search model. Electron densities were fitted using Coot (34) and refined in the CCP4 suite (35) using REFMAC5 (36) . Structural superposition and figures were prepared using PyMOL Molecular Graphics System (Schrödinger, LLC). Table-1 summarizes data collection and refinement statistics.',\n",
       "   'cite_spans': [{'start': 77,\n",
       "     'end': 81,\n",
       "     'text': '(30)',\n",
       "     'ref_id': 'BIBREF29'},\n",
       "    {'start': 203, 'end': 207, 'text': '(31)', 'ref_id': 'BIBREF30'},\n",
       "    {'start': 368, 'end': 372, 'text': '(34)', 'ref_id': 'BIBREF33'},\n",
       "    {'start': 422, 'end': 426, 'text': '(36)', 'ref_id': 'BIBREF35'}],\n",
       "   'ref_spans': [{'start': 538,\n",
       "     'end': 545,\n",
       "     'text': 'Table-1',\n",
       "     'ref_id': 'TABREF1'}],\n",
       "   'section': 'In Vitro Fluorescence IN Activity Assays'},\n",
       "  {'text': 'Analyses of interactions were performed using COCOMAPS (37) and LigPlot + (38) . Lysine Crosslinking: Homobifunctional BS 3 (Pierce, 2.5 mM) was added to full-length IN (20 µM) that was initially mixed with increasing concentrations of inhibitory peptide or Y3molecule (4-Acetamido-5-hydroxy-2,7-naphthalenedisulfonic acid, (Sigma, S981680)). Reactions were incubated for 30 minutes at room temperature, and quenched by adding Tris-HCl (pH 7.5) to a final concentration of 10 mM and analyzed on SDS-PAGE. Affinity Measurements: IN proteins were labeled with Monolith NT protein labeling kit BLUE-NHS (NanoTemper Technologies) and eluted in microscale thermophoresis (MST) buffer (50 mM Tris-HCl pH 7.0, 300 mM NaCl and 0.05% Tween-20). Labeled IN proteins were incubated with serial dilutions of inhibitor peptide, and MST measurements were carried out using the Monolith NT.115 in standard capillaries with 40% LED and 20% MST power.',\n",
       "   'cite_spans': [{'start': 55,\n",
       "     'end': 59,\n",
       "     'text': '(37)',\n",
       "     'ref_id': 'BIBREF36'},\n",
       "    {'start': 74, 'end': 78, 'text': '(38)', 'ref_id': 'BIBREF37'}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Structural Analysis:'},\n",
       "  {'text': 'Atomic coordinates and structure factors were deposited in the Protein Data Bank with the accession number 6EX9. . CTD of 1EX4 is modeled from a symmetry mate. Site-1 (S1) of all CCDCTD structures is clearly exposed and therefore a peptide was not superposed at S1. The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/301721 doi: bioRxiv preprint',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Accession numbers'}],\n",
       " 'bib_entries': {'BIBREF0': {'ref_id': 'b0',\n",
       "   'title': 'HIV drug resistance against strand transfer integrase inhibitors',\n",
       "   'authors': [{'first': 'K', 'middle': [], 'last': 'Anstett', 'suffix': ''},\n",
       "    {'first': 'B', 'middle': [], 'last': 'Brenner', 'suffix': ''},\n",
       "    {'first': 'T', 'middle': [], 'last': 'Mesplede', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': ['A'], 'last': 'Wainberg', 'suffix': ''}],\n",
       "   'year': 2017,\n",
       "   'venue': 'Retrovirology',\n",
       "   'volume': '14',\n",
       "   'issn': '',\n",
       "   'pages': '',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF1': {'ref_id': 'b1',\n",
       "   'title': 'Retroviral integrase protein and intasome nucleoprotein complex structures',\n",
       "   'authors': [{'first': 'J',\n",
       "     'middle': [],\n",
       "     'last': 'Grawenhoff',\n",
       "     'suffix': ''},\n",
       "    {'first': 'A', 'middle': ['N'], 'last': 'Engelman', 'suffix': ''}],\n",
       "   'year': 2017,\n",
       "   'venue': 'World J Biol Chem',\n",
       "   'volume': '8',\n",
       "   'issn': '',\n",
       "   'pages': '32--44',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF2': {'ref_id': 'b2',\n",
       "   'title': 'Identification of Phe187 as a crucial dimerization determinant facilitates crystallization of a monomeric retroviral integrase core domain',\n",
       "   'authors': [{'first': 'M', 'middle': [], 'last': 'Galilee', 'suffix': ''},\n",
       "    {'first': 'Alian', 'middle': [], 'last': '', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': [], 'last': '', 'suffix': ''}],\n",
       "   'year': 2014,\n",
       "   'venue': 'Structure',\n",
       "   'volume': '22',\n",
       "   'issn': '',\n",
       "   'pages': '1512--1519',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF3': {'ref_id': 'b3',\n",
       "   'title': 'A supramolecular assembly mediates lentiviral DNA integration',\n",
       "   'authors': [{'first': 'A',\n",
       "     'middle': [],\n",
       "     'last': 'Ballandras-Colas',\n",
       "     'suffix': ''},\n",
       "    {'first': 'D', 'middle': ['P'], 'last': 'Maskell', 'suffix': ''},\n",
       "    {'first': 'E', 'middle': [], 'last': 'Serrao', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': [], 'last': 'Locke', 'suffix': ''},\n",
       "    {'first': 'P', 'middle': [], 'last': 'Swuec', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': ['R'], 'last': 'Jonsson', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': [], 'last': 'Kotecha', 'suffix': ''},\n",
       "    {'first': 'N', 'middle': ['J'], 'last': 'Cook', 'suffix': ''},\n",
       "    {'first': 'V', 'middle': ['E'], 'last': 'Pye', 'suffix': ''},\n",
       "    {'first': 'I', 'middle': ['A'], 'last': 'Taylor', 'suffix': ''},\n",
       "    {'first': 'V', 'middle': [], 'last': 'Andresdottir', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': ['N'], 'last': 'Engelman', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': [], 'last': 'Costa', 'suffix': ''},\n",
       "    {'first': 'P', 'middle': [], 'last': 'Cherepanov', 'suffix': ''}],\n",
       "   'year': 2017,\n",
       "   'venue': 'Science',\n",
       "   'volume': '355',\n",
       "   'issn': '',\n",
       "   'pages': '93--95',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF4': {'ref_id': 'b4',\n",
       "   'title': 'Cryo-EM structures and atomic model of the HIV-1 strand transfer complex intasome',\n",
       "   'authors': [{'first': 'D', 'middle': ['O'], 'last': 'Passos', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': [], 'last': 'Li', 'suffix': ''},\n",
       "    {'first': 'R', 'middle': [], 'last': 'Yang', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': ['V'], 'last': 'Rebensburg', 'suffix': ''},\n",
       "    {'first': 'R', 'middle': [], 'last': 'Ghirlando', 'suffix': ''},\n",
       "    {'first': 'Y', 'middle': [], 'last': 'Jeon', 'suffix': ''},\n",
       "    {'first': 'N', 'middle': [], 'last': 'Shkriabai', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': [], 'last': 'Kvaratskhelia', 'suffix': ''},\n",
       "    {'first': 'R', 'middle': [], 'last': 'Craigie', 'suffix': ''},\n",
       "    {'first': 'D', 'middle': [], 'last': 'Lyumkis', 'suffix': ''}],\n",
       "   'year': 2017,\n",
       "   'venue': 'Science',\n",
       "   'volume': '355',\n",
       "   'issn': '',\n",
       "   'pages': '89--92',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF5': {'ref_id': 'b5',\n",
       "   'title': 'Cryo-EM reveals a novel octameric integrase structure for betaretroviral intasome function',\n",
       "   'authors': [{'first': 'A',\n",
       "     'middle': [],\n",
       "     'last': 'Ballandras-Colas',\n",
       "     'suffix': ''},\n",
       "    {'first': 'M', 'middle': [], 'last': 'Brown', 'suffix': ''},\n",
       "    {'first': 'N', 'middle': ['J'], 'last': 'Cook', 'suffix': ''},\n",
       "    {'first': 'T', 'middle': ['G'], 'last': 'Dewdney', 'suffix': ''},\n",
       "    {'first': 'B', 'middle': [], 'last': 'Demeler', 'suffix': ''},\n",
       "    {'first': 'P', 'middle': [], 'last': 'Cherepanov', 'suffix': ''},\n",
       "    {'first': 'D', 'middle': [], 'last': 'Lyumkis', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': ['N'], 'last': 'Engelman', 'suffix': ''}],\n",
       "   'year': 2016,\n",
       "   'venue': 'Nature',\n",
       "   'volume': '530',\n",
       "   'issn': '',\n",
       "   'pages': '358--361',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF6': {'ref_id': 'b6',\n",
       "   'title': 'Crystal structure of the Rous sarcoma virus intasome',\n",
       "   'authors': [{'first': 'Z', 'middle': [], 'last': 'Yin', 'suffix': ''},\n",
       "    {'first': 'K', 'middle': [], 'last': 'Shi', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': [], 'last': 'Banerjee', 'suffix': ''},\n",
       "    {'first': 'K', 'middle': ['K'], 'last': 'Pandey', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': [], 'last': 'Bera', 'suffix': ''},\n",
       "    {'first': 'D', 'middle': ['P'], 'last': 'Grandgenett', 'suffix': ''},\n",
       "    {'first': 'Aihara', 'middle': [], 'last': '', 'suffix': ''},\n",
       "    {'first': 'H', 'middle': [], 'last': '', 'suffix': ''}],\n",
       "   'year': 2016,\n",
       "   'venue': 'Nature',\n",
       "   'volume': '530',\n",
       "   'issn': '',\n",
       "   'pages': '362--366',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF7': {'ref_id': 'b7',\n",
       "   'title': 'The mechanism of retroviral integration from X-ray structures of its key intermediates',\n",
       "   'authors': [{'first': 'G',\n",
       "     'middle': ['N'],\n",
       "     'last': 'Maertens',\n",
       "     'suffix': ''},\n",
       "    {'first': 'S', 'middle': [], 'last': 'Hare', 'suffix': ''},\n",
       "    {'first': 'P', 'middle': [], 'last': 'Cherepanov', 'suffix': ''}],\n",
       "   'year': 2010,\n",
       "   'venue': 'Nature',\n",
       "   'volume': '468',\n",
       "   'issn': '',\n",
       "   'pages': '326--329',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF8': {'ref_id': 'b8',\n",
       "   'title': 'The Preserved HTH-Docking Cleft of HIV-1 Integrase Is Functionally Critical',\n",
       "   'authors': [{'first': 'M', 'middle': [], 'last': 'Galilee', 'suffix': ''},\n",
       "    {'first': 'E', 'middle': [], 'last': 'Britan-Rosich', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': ['L'], 'last': 'Griner', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': [], 'last': 'Uysal', 'suffix': ''},\n",
       "    {'first': 'V', 'middle': [], 'last': 'Baumgartel', 'suffix': ''},\n",
       "    {'first': 'D', 'middle': ['C'], 'last': 'Lamb', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': ['A'], 'last': 'Kossiakoff', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': [], 'last': 'Kotler', 'suffix': ''},\n",
       "    {'first': 'R', 'middle': ['M'], 'last': 'Stroud', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': [], 'last': 'Marx', 'suffix': ''},\n",
       "    {'first': 'Alian', 'middle': [], 'last': '', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': [], 'last': '', 'suffix': ''}],\n",
       "   'year': 2016,\n",
       "   'venue': 'Structure',\n",
       "   'volume': '24',\n",
       "   'issn': '',\n",
       "   'pages': '1936--1946',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF9': {'ref_id': 'b9',\n",
       "   'title': 'Resistance to pyridine-based inhibitor KF116 reveals an unexpected role of integrase in HIV-1 Gag-Pol proteolytic processing',\n",
       "   'authors': [{'first': 'A', 'middle': ['C'], 'last': 'Hoyte', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': ['V'], 'last': 'Jamin', 'suffix': ''},\n",
       "    {'first': 'P', 'middle': ['C'], 'last': 'Koneru', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': ['J'], 'last': 'Kobe', 'suffix': ''},\n",
       "    {'first': 'R', 'middle': ['C'], 'last': 'Larue', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': ['R'], 'last': 'Fuchs', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': ['N'], 'last': 'Engelman', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': [], 'last': 'Kvaratskhelia', 'suffix': ''}],\n",
       "   'year': 2017,\n",
       "   'venue': 'The Journal of biological chemistry',\n",
       "   'volume': '',\n",
       "   'issn': '',\n",
       "   'pages': '',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF10': {'ref_id': 'b10',\n",
       "   'title': 'Small-molecule inhibitors of the LEDGF/p75 binding site of integrase block HIV replication and modulate integrase multimerization',\n",
       "   'authors': [{'first': 'F', 'middle': [], 'last': 'Christ', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': [], 'last': 'Shaw', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': [], 'last': 'Demeulemeester', 'suffix': ''},\n",
       "    {'first': 'B', 'middle': ['A'], 'last': 'Desimmie', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': [], 'last': 'Marchand', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': [], 'last': 'Butler', 'suffix': ''},\n",
       "    {'first': 'W', 'middle': [], 'last': 'Smets', 'suffix': ''},\n",
       "    {'first': 'P', 'middle': [], 'last': 'Chaltin', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': [], 'last': 'Westby', 'suffix': ''},\n",
       "    {'first': 'Z', 'middle': [], 'last': 'Debyser', 'suffix': ''},\n",
       "    {'first': 'C', 'middle': [], 'last': 'Pickford', 'suffix': ''}],\n",
       "   'year': 2012,\n",
       "   'venue': 'Antimicrob Agents Chemother',\n",
       "   'volume': '56',\n",
       "   'issn': '',\n",
       "   'pages': '4365--4374',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF11': {'ref_id': 'b11',\n",
       "   'title': 'Allosteric inhibition of human immunodeficiency virus integrase: late block during viral replication and abnormal multimerization involving specific protein domains',\n",
       "   'authors': [{'first': 'K', 'middle': [], 'last': 'Gupta', 'suffix': ''},\n",
       "    {'first': 'T', 'middle': [], 'last': 'Brady', 'suffix': ''},\n",
       "    {'first': 'B', 'middle': ['M'], 'last': 'Dyer', 'suffix': ''},\n",
       "    {'first': 'N', 'middle': [], 'last': 'Malani', 'suffix': ''},\n",
       "    {'first': 'Y', 'middle': [], 'last': 'Hwang', 'suffix': ''},\n",
       "    {'first': 'F', 'middle': [], 'last': 'Male', 'suffix': ''},\n",
       "    {'first': 'R', 'middle': ['T'], 'last': 'Nolte', 'suffix': ''},\n",
       "    {'first': 'L', 'middle': [], 'last': 'Wang', 'suffix': ''},\n",
       "    {'first': 'E', 'middle': [], 'last': 'Velthuisen', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': [], 'last': 'Jeffrey', 'suffix': ''},\n",
       "    {'first': 'G', 'middle': ['D'], 'last': 'Van Duyne', 'suffix': ''},\n",
       "    {'first': 'F', 'middle': ['D'], 'last': 'Bushman', 'suffix': ''}],\n",
       "   'year': 2014,\n",
       "   'venue': 'The Journal of biological chemistry',\n",
       "   'volume': '289',\n",
       "   'issn': '',\n",
       "   'pages': '20477--20488',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF12': {'ref_id': 'b12',\n",
       "   'title': 'HIV-1 integrase multimerization as a therapeutic target',\n",
       "   'authors': [{'first': 'L', 'middle': [], 'last': 'Feng', 'suffix': ''},\n",
       "    {'first': 'R', 'middle': ['C'], 'last': 'Larue', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': [], 'last': 'Slaughter', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': ['J'], 'last': 'Kessl', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': [], 'last': 'Kvaratskhelia', 'suffix': ''}],\n",
       "   'year': 2015,\n",
       "   'venue': 'Curr Top Microbiol Immunol',\n",
       "   'volume': '389',\n",
       "   'issn': '',\n",
       "   'pages': '93--119',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF13': {'ref_id': 'b13',\n",
       "   'title': 'Allosteric integrase inhibitor potency is determined through the inhibition of HIV-1 particle maturation',\n",
       "   'authors': [{'first': 'K', 'middle': ['A'], 'last': 'Jurado', 'suffix': ''},\n",
       "    {'first': 'H', 'middle': [], 'last': 'Wang', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': [], 'last': 'Slaughter', 'suffix': ''},\n",
       "    {'first': 'L', 'middle': [], 'last': 'Feng', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': ['J'], 'last': 'Kessl', 'suffix': ''},\n",
       "    {'first': 'Y', 'middle': [], 'last': 'Koh', 'suffix': ''},\n",
       "    {'first': 'W', 'middle': [], 'last': 'Wang', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': [], 'last': 'Ballandras-Colas', 'suffix': ''},\n",
       "    {'first': 'P', 'middle': ['A'], 'last': 'Patel', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': ['R'], 'last': 'Fuchs', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': [], 'last': 'Kvaratskhelia', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': [], 'last': 'Engelman', 'suffix': ''}],\n",
       "   'year': 2013,\n",
       "   'venue': 'Proceedings of the National Academy of Sciences of the United States of America',\n",
       "   'volume': '110',\n",
       "   'issn': '',\n",
       "   'pages': '8690--8695',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF14': {'ref_id': 'b14',\n",
       "   'title': 'Discovery of small molecule HIV-1 integrase dimerization inhibitors',\n",
       "   'authors': [{'first': 'C', 'middle': [], 'last': 'Tintori', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': [], 'last': 'Demeulemeester', 'suffix': ''},\n",
       "    {'first': 'L', 'middle': [], 'last': 'Franchi', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': [], 'last': 'Massa', 'suffix': ''},\n",
       "    {'first': 'Z', 'middle': [], 'last': 'Debyser', 'suffix': ''},\n",
       "    {'first': 'F', 'middle': [], 'last': 'Christ', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': [], 'last': 'Botta', 'suffix': ''}],\n",
       "   'year': 2012,\n",
       "   'venue': 'Bioorganic & medicinal chemistry letters',\n",
       "   'volume': '22',\n",
       "   'issn': '',\n",
       "   'pages': '3109--3114',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF15': {'ref_id': 'b15',\n",
       "   'title': 'Structure of the catalytic domain of avian sarcoma virus integrase with a bound HIV-1 integrase-targeted inhibitor',\n",
       "   'authors': [{'first': 'J', 'middle': [], 'last': 'Lubkowski', 'suffix': ''},\n",
       "    {'first': 'F', 'middle': [], 'last': 'Yang', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': [], 'last': 'Alexandratos', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': [], 'last': 'Wlodawer', 'suffix': ''},\n",
       "    {'first': 'H', 'middle': [], 'last': 'Zhao', 'suffix': ''},\n",
       "    {'first': 'T', 'middle': ['R'], 'last': 'Burke', 'suffix': ''},\n",
       "    {'first': '', 'middle': [], 'last': 'Jr', 'suffix': ''},\n",
       "    {'first': 'N', 'middle': [], 'last': 'Neamati', 'suffix': ''},\n",
       "    {'first': 'Y', 'middle': [], 'last': 'Pommier', 'suffix': ''},\n",
       "    {'first': 'G', 'middle': [], 'last': 'Merkel', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': ['M'], 'last': 'Skalka', 'suffix': ''}],\n",
       "   'year': 1998,\n",
       "   'venue': 'Proceedings of the National Academy of Sciences of the United States of America',\n",
       "   'volume': '95',\n",
       "   'issn': '',\n",
       "   'pages': '4831--4836',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF16': {'ref_id': 'b16',\n",
       "   'title': 'Crystal structure of the HIV-1 integrase core domain in complex with sucrose reveals details of an allosteric inhibitory binding site',\n",
       "   'authors': [{'first': 'J', 'middle': [], 'last': 'Wielens', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': ['J'], 'last': 'Headey', 'suffix': ''},\n",
       "    {'first': 'D', 'middle': [], 'last': 'Jeevarajah', 'suffix': ''},\n",
       "    {'first': 'D', 'middle': ['I'], 'last': 'Rhodes', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': [], 'last': 'Deadman', 'suffix': ''},\n",
       "    {'first': 'D', 'middle': ['K'], 'last': 'Chalmers', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': ['J'], 'last': 'Scanlon', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': ['W'], 'last': 'Parker', 'suffix': ''}],\n",
       "   'year': 2010,\n",
       "   'venue': 'FEBS Lett',\n",
       "   'volume': '584',\n",
       "   'issn': '',\n",
       "   'pages': '1455--1462',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF17': {'ref_id': 'b17',\n",
       "   'title': 'Identification of an inhibitor-binding site to HIV-1 integrase with affinity acetylation and mass spectrometry',\n",
       "   'authors': [{'first': 'N', 'middle': [], 'last': 'Shkriabai', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': ['S'], 'last': 'Patil', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': [], 'last': 'Hess', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': ['R'], 'last': 'Budihas', 'suffix': ''},\n",
       "    {'first': 'R', 'middle': [], 'last': 'Craigie', 'suffix': ''},\n",
       "    {'first': 'T', 'middle': ['R'], 'last': 'Burke', 'suffix': ''},\n",
       "    {'first': '', 'middle': [], 'last': 'Jr', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': ['F'], 'last': 'Le Grice', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': [], 'last': 'Kvaratskhelia', 'suffix': ''}],\n",
       "   'year': 2004,\n",
       "   'venue': 'Proceedings of the National Academy of Sciences of the United States of America',\n",
       "   'volume': '101',\n",
       "   'issn': '',\n",
       "   'pages': '6894--6899',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF18': {'ref_id': 'b18',\n",
       "   'title': 'as a New Allosteric HIV-1 Integrase Inhibitor: Molecular Modeling and Biological Evaluation',\n",
       "   'authors': [{'first': 'F', 'middle': [], 'last': 'Esposito', 'suffix': ''},\n",
       "    {'first': 'C', 'middle': [], 'last': 'Tintori', 'suffix': ''},\n",
       "    {'first': 'R', 'middle': [], 'last': 'Martini', 'suffix': ''},\n",
       "    {'first': 'F', 'middle': [], 'last': 'Christ', 'suffix': ''},\n",
       "    {'first': 'Z', 'middle': [], 'last': 'Debyser', 'suffix': ''},\n",
       "    {'first': 'R', 'middle': [], 'last': 'Ferrarese', 'suffix': ''},\n",
       "    {'first': 'G', 'middle': [], 'last': 'Cabiddu', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': [], 'last': 'Corona', 'suffix': ''},\n",
       "    {'first': 'E', 'middle': ['R'], 'last': 'Ceresola', 'suffix': ''}],\n",
       "   'year': None,\n",
       "   'venue': '',\n",
       "   'volume': '16',\n",
       "   'issn': '',\n",
       "   'pages': '2507--2512',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF19': {'ref_id': 'b19',\n",
       "   'title': 'Fragment-based design of ligands targeting a novel site on the integrase enzyme of human immunodeficiency virus 1',\n",
       "   'authors': [{'first': 'J', 'middle': [], 'last': 'Wielens', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': ['J'], 'last': 'Headey', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': ['J'], 'last': 'Deadman', 'suffix': ''},\n",
       "    {'first': 'D', 'middle': ['I'], 'last': 'Rhodes', 'suffix': ''},\n",
       "    {'first': 'G', 'middle': ['T'], 'last': 'Le', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': ['W'], 'last': 'Parker', 'suffix': ''},\n",
       "    {'first': 'D', 'middle': ['K'], 'last': 'Chalmers', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': ['J'], 'last': 'Scanlon', 'suffix': ''}],\n",
       "   'year': 2011,\n",
       "   'venue': 'ChemMedChem',\n",
       "   'volume': '6',\n",
       "   'issn': '',\n",
       "   'pages': '258--261',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF20': {'ref_id': 'b20',\n",
       "   'title': 'Structural basis for a new mechanism of inhibition of HIV-1 integrase identified by fragment screening and structure-based design',\n",
       "   'authors': [{'first': 'D', 'middle': ['I'], 'last': 'Rhodes', 'suffix': ''},\n",
       "    {'first': 'T', 'middle': ['S'], 'last': 'Peat', 'suffix': ''},\n",
       "    {'first': 'N', 'middle': [], 'last': 'Vandegraaff', 'suffix': ''},\n",
       "    {'first': 'D', 'middle': [], 'last': 'Jeevarajah', 'suffix': ''},\n",
       "    {'first': 'G', 'middle': [], 'last': 'Le', 'suffix': ''},\n",
       "    {'first': 'E', 'middle': ['D'], 'last': 'Jones', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': ['A'], 'last': 'Smith', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': ['A'], 'last': 'Coates', 'suffix': ''},\n",
       "    {'first': 'L', 'middle': ['J'], 'last': 'Winfield', 'suffix': ''},\n",
       "    {'first': 'N', 'middle': [], 'last': 'Thienthong', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': [], 'last': 'Newman', 'suffix': ''},\n",
       "    {'first': 'D', 'middle': [], 'last': 'Lucent', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': ['H'], 'last': 'Ryan', 'suffix': ''},\n",
       "    {'first': 'G', 'middle': ['P'], 'last': 'Savage', 'suffix': ''},\n",
       "    {'first': 'C', 'middle': ['L'], 'last': 'Francis', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': ['J'], 'last': 'Deadman', 'suffix': ''}],\n",
       "   'year': 2011,\n",
       "   'venue': 'Antivir Chem Chemother',\n",
       "   'volume': '21',\n",
       "   'issn': '',\n",
       "   'pages': '155--168',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF21': {'ref_id': 'b21',\n",
       "   'title': 'Discovery of a Distinct Chemical and Mechanistic Class of Allosteric HIV-1 Integrase Inhibitors with Antiretroviral Activity',\n",
       "   'authors': [{'first': 'C', 'middle': [], 'last': 'Burlein', 'suffix': ''},\n",
       "    {'first': 'C', 'middle': [], 'last': 'Wang', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': [], 'last': 'Xu', 'suffix': ''},\n",
       "    {'first': 'T', 'middle': [], 'last': 'Bhatt', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': [], 'last': 'Stahlhut', 'suffix': ''},\n",
       "    {'first': 'Y', 'middle': [], 'last': 'Ou', 'suffix': ''},\n",
       "    {'first': 'G', 'middle': ['C'], 'last': 'Adam', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': [], 'last': 'Heath', 'suffix': ''},\n",
       "    {'first': 'D', 'middle': ['J'], 'last': 'Klein', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': [], 'last': 'Sanders', 'suffix': ''},\n",
       "    {'first': 'K', 'middle': [], 'last': 'Narayan', 'suffix': ''},\n",
       "    {'first': 'P', 'middle': [], 'last': 'Abeywickrema', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': ['R'], 'last': 'Heo', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': ['S'], 'last': 'Carroll', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': ['A'], 'last': 'Grobler', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': [], 'last': 'Sharma', 'suffix': ''},\n",
       "    {'first': 'T', 'middle': ['L'], 'last': 'Diamond', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': [], 'last': 'Converso', 'suffix': ''},\n",
       "    {'first': 'D', 'middle': ['J'], 'last': 'Krosky', 'suffix': ''}],\n",
       "   'year': 2017,\n",
       "   'venue': 'ACS Chem Biol',\n",
       "   'volume': '12',\n",
       "   'issn': '',\n",
       "   'pages': '2858--2865',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF22': {'ref_id': 'b22',\n",
       "   'title': 'Genetic analyses of conserved residues in the carboxyl-terminal domain of human immunodeficiency virus type 1 integrase',\n",
       "   'authors': [{'first': 'R', 'middle': [], 'last': 'Lu', 'suffix': ''},\n",
       "    {'first': 'H', 'middle': ['Z'], 'last': 'Ghory', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': [], 'last': 'Engelman', 'suffix': ''}],\n",
       "   'year': 2005,\n",
       "   'venue': 'Journal of virology',\n",
       "   'volume': '79',\n",
       "   'issn': '',\n",
       "   'pages': '10356--10368',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF23': {'ref_id': 'b23',\n",
       "   'title': 'Catalytically-active complex of HIV-1 integrase with a viral DNA substrate binds anti-integrase drugs',\n",
       "   'authors': [{'first': 'A', 'middle': [], 'last': 'Alian', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': ['L'], 'last': 'Griner', 'suffix': ''},\n",
       "    {'first': 'V', 'middle': [], 'last': 'Chiang', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': [], 'last': 'Tsiang', 'suffix': ''},\n",
       "    {'first': 'G', 'middle': [], 'last': 'Jones', 'suffix': ''},\n",
       "    {'first': 'G', 'middle': [], 'last': 'Birkus', 'suffix': ''},\n",
       "    {'first': 'R', 'middle': [], 'last': 'Geleziunas', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': ['D'], 'last': 'Leavitt', 'suffix': ''},\n",
       "    {'first': 'R', 'middle': ['M'], 'last': 'Stroud', 'suffix': ''}],\n",
       "   'year': 2009,\n",
       "   'venue': 'Proceedings of the National Academy of Sciences of the United States of America',\n",
       "   'volume': '106',\n",
       "   'issn': '',\n",
       "   'pages': '8192--8197',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF24': {'ref_id': 'b24',\n",
       "   'title': 'Characterization of feline immunodeficiency virus integrase and analysis of functional domains',\n",
       "   'authors': [{'first': 'Y', 'middle': [], 'last': 'Shibagaki', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': ['L'], 'last': 'Holmes', 'suffix': ''},\n",
       "    {'first': 'R', 'middle': ['S'], 'last': 'Appa', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': ['A'], 'last': 'Chow', 'suffix': ''}],\n",
       "   'year': 1997,\n",
       "   'venue': 'Virology',\n",
       "   'volume': '230',\n",
       "   'issn': '',\n",
       "   'pages': '1--10',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF25': {'ref_id': 'b25',\n",
       "   'title': 'Identification of discrete functional domains of HIV-1 integrase and their organization within an active multimeric complex',\n",
       "   'authors': [{'first': 'A', 'middle': [], 'last': 'Engelman', 'suffix': ''},\n",
       "    {'first': 'F', 'middle': ['D'], 'last': 'Bushman', 'suffix': ''},\n",
       "    {'first': 'Craigie', 'middle': [], 'last': '', 'suffix': ''},\n",
       "    {'first': 'R', 'middle': [], 'last': '', 'suffix': ''}],\n",
       "   'year': 1993,\n",
       "   'venue': 'EMBO J',\n",
       "   'volume': '12',\n",
       "   'issn': '',\n",
       "   'pages': '3269--3275',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF26': {'ref_id': 'b26',\n",
       "   'title': 'A critical role of the C-terminal segment for allosteric inhibitor-induced aberrant multimerization of HIV-1 integrase',\n",
       "   'authors': [{'first': 'N', 'middle': [], 'last': 'Shkriabai', 'suffix': ''},\n",
       "    {'first': 'V', 'middle': [], 'last': 'Dharmarajan', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': [], 'last': 'Slaughter', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': ['J'], 'last': 'Kessl', 'suffix': ''},\n",
       "    {'first': 'R', 'middle': ['C'], 'last': 'Larue', 'suffix': ''},\n",
       "    {'first': 'L', 'middle': [], 'last': 'Feng', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': ['R'], 'last': 'Fuchs', 'suffix': ''},\n",
       "    {'first': 'P', 'middle': ['R'], 'last': 'Griffin', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': [], 'last': 'Kvaratskhelia', 'suffix': ''}],\n",
       "   'year': 2014,\n",
       "   'venue': 'The Journal of biological chemistry',\n",
       "   'volume': '289',\n",
       "   'issn': '',\n",
       "   'pages': '26430--26440',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF27': {'ref_id': 'b27',\n",
       "   'title': 'Structural Basis for Inhibitor-Induced Aggregation of HIV Integrase',\n",
       "   'authors': [{'first': 'K', 'middle': [], 'last': 'Gupta', 'suffix': ''},\n",
       "    {'first': 'V', 'middle': [], 'last': 'Turkki', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': [], 'last': 'Sherrill-Mix', 'suffix': ''},\n",
       "    {'first': 'Y', 'middle': [], 'last': 'Hwang', 'suffix': ''},\n",
       "    {'first': 'G', 'middle': [], 'last': 'Eilers', 'suffix': ''},\n",
       "    {'first': 'L', 'middle': [], 'last': 'Taylor', 'suffix': ''},\n",
       "    {'first': 'C', 'middle': [], 'last': 'Mcdanal', 'suffix': ''},\n",
       "    {'first': 'P', 'middle': [], 'last': 'Wang', 'suffix': ''},\n",
       "    {'first': 'D', 'middle': [], 'last': 'Temelkoff', 'suffix': ''},\n",
       "    {'first': 'R', 'middle': ['T'], 'last': 'Nolte', 'suffix': ''},\n",
       "    {'first': 'E', 'middle': [], 'last': 'Velthuisen', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': [], 'last': 'Jeffrey', 'suffix': ''},\n",
       "    {'first': 'G', 'middle': ['D'], 'last': 'Van Duyne', 'suffix': ''},\n",
       "    {'first': 'F', 'middle': ['D'], 'last': 'Bushman', 'suffix': ''}],\n",
       "   'year': 2016,\n",
       "   'venue': 'PLoS Biol',\n",
       "   'volume': '14',\n",
       "   'issn': '',\n",
       "   'pages': '',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF28': {'ref_id': 'b28',\n",
       "   'title': 'In vitro functional analyses of the human immunodeficiency virus type 1 (HIV-1) integrase mutants give new insights into the intasome assembly',\n",
       "   'authors': [{'first': 'C', 'middle': [], 'last': 'Cellier', 'suffix': ''},\n",
       "    {'first': 'K', 'middle': [], 'last': 'Moreau', 'suffix': ''},\n",
       "    {'first': 'K', 'middle': [], 'last': 'Gallay', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': [], 'last': 'Ballandras', 'suffix': ''},\n",
       "    {'first': 'P', 'middle': [], 'last': 'Gouet', 'suffix': ''},\n",
       "    {'first': 'C', 'middle': [], 'last': 'Ronfort', 'suffix': ''}],\n",
       "   'year': 2013,\n",
       "   'venue': 'Virology',\n",
       "   'volume': '439',\n",
       "   'issn': '',\n",
       "   'pages': '97--104',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF29': {'ref_id': 'b29',\n",
       "   'title': 'Oligonucleotide-based assays for integrase activity',\n",
       "   'authors': [{'first': 'G', 'middle': [], 'last': 'Merkel', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': ['D'], 'last': 'Andrake', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': [], 'last': 'Ramcharan', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': ['M'], 'last': 'Skalka', 'suffix': ''}],\n",
       "   'year': 2009,\n",
       "   'venue': 'Methods',\n",
       "   'volume': '47',\n",
       "   'issn': '',\n",
       "   'pages': '243--248',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF30': {'ref_id': 'b30',\n",
       "   'title': 'iMOSFLM: a new graphical interface for diffraction-image processing with MOSFLM',\n",
       "   'authors': [{'first': 'T', 'middle': ['G'], 'last': 'Battye', 'suffix': ''},\n",
       "    {'first': 'L', 'middle': [], 'last': 'Kontogiannis', 'suffix': ''},\n",
       "    {'first': 'O', 'middle': [], 'last': 'Johnson', 'suffix': ''},\n",
       "    {'first': 'H', 'middle': ['R'], 'last': 'Powell', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': ['G'], 'last': 'Leslie', 'suffix': ''}],\n",
       "   'year': 2011,\n",
       "   'venue': 'Acta crystallographica. Section D, Biological crystallography',\n",
       "   'volume': '67',\n",
       "   'issn': '',\n",
       "   'pages': '271--281',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF31': {'ref_id': 'b31',\n",
       "   'title': 'Phaser crystallographic software',\n",
       "   'authors': [{'first': 'A', 'middle': ['J'], 'last': 'Mccoy', 'suffix': ''},\n",
       "    {'first': 'R', 'middle': ['W'], 'last': 'Grosse-Kunstleve', 'suffix': ''},\n",
       "    {'first': 'P', 'middle': ['D'], 'last': 'Adams', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': ['D'], 'last': 'Winn', 'suffix': ''},\n",
       "    {'first': 'L', 'middle': ['C'], 'last': 'Storoni', 'suffix': ''},\n",
       "    {'first': 'R', 'middle': ['J'], 'last': 'Read', 'suffix': ''}],\n",
       "   'year': 2007,\n",
       "   'venue': 'J Appl Crystallogr',\n",
       "   'volume': '40',\n",
       "   'issn': '',\n",
       "   'pages': '658--674',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF32': {'ref_id': 'b32',\n",
       "   'title': 'Three new structures of the core domain of HIV-1 integrase: an active site that binds magnesium',\n",
       "   'authors': [{'first': 'Y', 'middle': [], 'last': 'Goldgur', 'suffix': ''},\n",
       "    {'first': 'F', 'middle': [], 'last': 'Dyda', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': ['B'], 'last': 'Hickman', 'suffix': ''},\n",
       "    {'first': 'T', 'middle': ['M'], 'last': 'Jenkins', 'suffix': ''},\n",
       "    {'first': 'R', 'middle': [], 'last': 'Craigie', 'suffix': ''},\n",
       "    {'first': 'D', 'middle': ['R'], 'last': 'Davies', 'suffix': ''}],\n",
       "   'year': 1998,\n",
       "   'venue': 'Proceedings of the National Academy of Sciences of the United States of America',\n",
       "   'volume': '95',\n",
       "   'issn': '',\n",
       "   'pages': '9150--9154',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF33': {'ref_id': 'b33',\n",
       "   'title': 'Features and development of Coot',\n",
       "   'authors': [{'first': 'P', 'middle': [], 'last': 'Emsley', 'suffix': ''},\n",
       "    {'first': 'B', 'middle': [], 'last': 'Lohkamp', 'suffix': ''},\n",
       "    {'first': 'W', 'middle': ['G'], 'last': 'Scott', 'suffix': ''},\n",
       "    {'first': 'K', 'middle': [], 'last': 'Cowtan', 'suffix': ''}],\n",
       "   'year': 2010,\n",
       "   'venue': 'Acta crystallographica. Section D, Biological crystallography',\n",
       "   'volume': '66',\n",
       "   'issn': '',\n",
       "   'pages': '486--501',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF34': {'ref_id': 'b34',\n",
       "   'title': 'Overview of the CCP4 suite and current developments',\n",
       "   'authors': [{'first': 'M', 'middle': ['D'], 'last': 'Winn', 'suffix': ''},\n",
       "    {'first': 'C', 'middle': ['C'], 'last': 'Ballard', 'suffix': ''},\n",
       "    {'first': 'K', 'middle': ['D'], 'last': 'Cowtan', 'suffix': ''},\n",
       "    {'first': 'E', 'middle': ['J'], 'last': 'Dodson', 'suffix': ''},\n",
       "    {'first': 'P', 'middle': [], 'last': 'Emsley', 'suffix': ''},\n",
       "    {'first': 'P', 'middle': ['R'], 'last': 'Evans', 'suffix': ''},\n",
       "    {'first': 'R', 'middle': ['M'], 'last': 'Keegan', 'suffix': ''},\n",
       "    {'first': 'E', 'middle': ['B'], 'last': 'Krissinel', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': ['G'], 'last': 'Leslie', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': [], 'last': 'Mccoy', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': ['J'], 'last': 'Mcnicholas', 'suffix': ''},\n",
       "    {'first': 'G', 'middle': ['N'], 'last': 'Murshudov', 'suffix': ''},\n",
       "    {'first': 'N', 'middle': ['S'], 'last': 'Pannu', 'suffix': ''},\n",
       "    {'first': 'E', 'middle': ['A'], 'last': 'Potterton', 'suffix': ''},\n",
       "    {'first': 'H', 'middle': ['R'], 'last': 'Powell', 'suffix': ''},\n",
       "    {'first': 'R', 'middle': ['J'], 'last': 'Read', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': [], 'last': 'Vagin', 'suffix': ''},\n",
       "    {'first': 'K', 'middle': ['S'], 'last': 'Wilson', 'suffix': ''}],\n",
       "   'year': 2011,\n",
       "   'venue': 'Acta crystallographica. Section D, Biological crystallography',\n",
       "   'volume': '67',\n",
       "   'issn': '',\n",
       "   'pages': '235--242',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF35': {'ref_id': 'b35',\n",
       "   'title': 'REFMAC5 for the refinement of macromolecular crystal structures',\n",
       "   'authors': [{'first': 'G',\n",
       "     'middle': ['N'],\n",
       "     'last': 'Murshudov',\n",
       "     'suffix': ''},\n",
       "    {'first': 'P', 'middle': [], 'last': 'Skubak', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': ['A'], 'last': 'Lebedev', 'suffix': ''},\n",
       "    {'first': 'N', 'middle': ['S'], 'last': 'Pannu', 'suffix': ''},\n",
       "    {'first': 'R', 'middle': ['A'], 'last': 'Steiner', 'suffix': ''},\n",
       "    {'first': 'R', 'middle': ['A'], 'last': 'Nicholls', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': ['D'], 'last': 'Winn', 'suffix': ''},\n",
       "    {'first': 'F', 'middle': [], 'last': 'Long', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': ['A'], 'last': 'Vagin', 'suffix': ''}],\n",
       "   'year': 2011,\n",
       "   'venue': 'Acta crystallographica. Section D, Biological crystallography',\n",
       "   'volume': '67',\n",
       "   'issn': '',\n",
       "   'pages': '355--367',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF36': {'ref_id': 'b36',\n",
       "   'title': 'COCOMAPS: a web application to analyze and visualize contacts at the interface of biomolecular complexes',\n",
       "   'authors': [{'first': 'A', 'middle': [], 'last': 'Vangone', 'suffix': ''},\n",
       "    {'first': 'R', 'middle': [], 'last': 'Spinelli', 'suffix': ''},\n",
       "    {'first': 'V', 'middle': [], 'last': 'Scarano', 'suffix': ''},\n",
       "    {'first': 'L', 'middle': [], 'last': 'Cavallo', 'suffix': ''},\n",
       "    {'first': 'R', 'middle': [], 'last': 'Oliva', 'suffix': ''}],\n",
       "   'year': 2011,\n",
       "   'venue': 'Bioinformatics',\n",
       "   'volume': '27',\n",
       "   'issn': '',\n",
       "   'pages': '2915--2916',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF37': {'ref_id': 'b37',\n",
       "   'title': 'LigPlot+: multiple ligand-protein interaction diagrams for drug discovery',\n",
       "   'authors': [{'first': 'R',\n",
       "     'middle': ['A'],\n",
       "     'last': 'Laskowski',\n",
       "     'suffix': ''},\n",
       "    {'first': 'M', 'middle': ['B'], 'last': 'Swindells', 'suffix': ''}],\n",
       "   'year': 2011,\n",
       "   'venue': 'J Chem Inf Model',\n",
       "   'volume': '51',\n",
       "   'issn': '',\n",
       "   'pages': '2778--2786',\n",
       "   'other_ids': {}}},\n",
       " 'ref_entries': {'FIGREF0': {'text': \". IN and truncation constructs were added to the reaction mixture to a final concentration of 4.5 µM. Domain combinations were preincubated at a ratio of 1:1 to a final concentration of 4.5 µM and then added to the reaction mixture containing 25 mM Tris-HCl (pH 7.5), 5 mM dithiothreitol and 50 nM DNA, and incubated for 15 minutes on ice. Next, MnCl 2 (10 mM) was added and the reactions were incubated at 37 °C for 1hr, and Biotin-CTACCGCATTAAAGCTTATCGCGGATA. IN and truncation constructs were added to reaction mixtures to a final concentration of 4.5 µM. Domain combinations were preincubated at a ratio of 1:1 to a final concentration of 4.5 µM, and then added to the reaction mixture containing 25 mM HEPES (pH 7.5), 5 mM dithiothreitol, 0.25 µM donor-DNA, 10 mM MnCl 2 and incubated for 15 minutes on ice. Next, acceptor-DNA was added (1.5 µM) and reactions were incubated at 37 °C for 1hr and quenched by adding EDTA (10 mM). Reaction products were purified using Streptavidin beads (GE Healthcare). For the inhibition assays of 3'-processing and strandtransfer activities, Fab (4.5 µM, 1:1 molar ratio) or peptide (70 µM, ~ 15:1 excess molar ratio to IN) were added prior to the addition of substrate DNA. Crystallization, Data Collection and Structure Determination: Diffracting-quality crystals of protein-peptide mixture (8.8 mg/ml protein supplemented with 1 mM peptide) were obtained in the JCSG-G7 condition (0.1 M Succinic acid pH 7.0, 15% (w/v) PEG-3350). Cryosolutions were supplemented with 20% (v/v) glycerol and 0.5 M NaCl. Diffraction data were collected at beamline ID23-1 of the European Synchrotron Radiation Facility (ESRF),\",\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF1': {'text': 'Allosteric sites of HIV-1 IN and the inhibition of IN activity. (A) Surface rendering of HIV-1 IN CCD dimer (PDB code: 3NF8) showing the various allosteric sites. LEDGIN (magenta sticks), FBP (blue sticks) and Y3 site bound to CDQ molecule (cyan sticks, PDB code: 3NF8) and Y3 molecule (orange sticks, modeled from PDB code: 1A5V). Sucrose (SUC, blue sticks, PDB code: 3L3V) is also modeled into the structure. Magnesium ion (red sphere) was superimposed from HIV-1 intasome (PDB code: 5U1C) to implicate the catalytic site. (B) Conservation of NTD (magenta) and CTD (orange) packing around α4 of CCD (green) at site-1. Structures are of HIV-1, MVV, RSV, MMTV and PFV intasomes (PDB codes: 5U1C, 5M0R, 5EJK, 3JCA, 3OS0, respectively). (C) Structure of IN CCD (green) bound to CDR3 and CDR2 of variable heavy (V H , blue) and light (V L , cyan) chains of Fab specific to site-1 (PDB code: 5EU7). Interacting Fab residues are shown with lines. F: finger loop. (D) IN catalytic activity as affected by Fab and the inhibitor peptide. Values are means of three repeats (± SEM). (E) Effect of peptide binding on IN multimerization. Values are means (± SEM) of band intensities from three SDS-PAGE gels resolving the various multimeric states of IN (right panel). Representative gel resolving the monomeric (M) form of uncross-linked sample of full length IN (lane 1) or cross-linked in the absence (lane 2) or presence of increasing concentrations of peptide (lanes 3-6). Molecular mass standards (kDa) are shown. Crystal structure of IN CCD-peptide complex. (A) Walleye stereo view of CCD (green) with the peptide and its symmetry mate bound at site-1 (S1) and site-2 (S2), respectively. Blue mesh represents electron density maps (F O -F C , 3.0σ) calculated after omitting the peptide residues (black). (B) NTD (magenta) and CTD (orange) docking at site-1 (S1) and CTD (blue) docking at site-2 (S2) within the same CCD α4 (green). F: finger loop. Red sphere: magnesium ion of catalytic site (PDB code: 5U1C). (C) Conservation of CTD (blue) packing against NTD linker (magenta) and α4 of CCD ) at site-2. Structures are of HIV-1, MVV, RSV, MMTV and PFV (PDB codes: 5U1C, 5M0R, 5EJK, 3JCA, 3OS0, respectively) intasomes. F: finger loop. Red sphere: magnesium ion of catalytic site (PDB code: 5U1C). (D) Analysis of peptide (black) interactions with the CCD (green) at site-1 (top panel) and site-2 (bottom). Interacting residues are shown with lines (peptide) or sticks (CCD). Dashed lines indicate direct hydrogen bonds (magenta) or mediated (cyan) by water molecules (cyan spheres). (E) Walleye stereo view of the HIV-1 intasome (PDB code: 5U1C) superimposed with the peptide (black) bound to site-1 (top) and site-2 (bottom). Red sphere indicate magnesium ion of catalytic site. (F) IN multimerization as affected by the Y3-molecule and is presented by the percentage of monomer formation. Values are means (± SEM) of band intensities from three SDS-PAGE gels resolving the various multimeric states of IN (right panel). Representative gel resolving the monomeric (M) form of uncross-linked sample of full length IN (lane 1) or cross-linked in the absence (lane 2) or presence of increasing concentrations of Y3-molecule (lanes 3-6). Molecular mass standards (kDa) are shown. Peptide binding affinities to IN truncation domains. (A) Superposition of the peptide (black), bound at site-1 (S1) and site-2 (S2), to NTDCCD structure (PDB code: 1K6Y). (B) Superposition of the peptide (black), bound at site-2 (S2), to the various CCDCTD structures of HIV-1 intasome (orange, PDB code: 5U1C) or apo proteins of HIV-1 (blue, PDB code: 1EX4), SIV (cyan, PDB code: 1C6V), RSV (yellow, PDB code: 1C1A), and MVV (magenta, PDB code: 5T3A)',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF2': {'text': \"(C) Peptide binding affinity to HIV-1 IN CCD, NTDCCD and CCDCTD. Values are means of three repeats (±SD). ΔF Norm : normalized fluorescence units (% of bound and unbound peptide). IN activity and multimerization. (A) 3'-processing activity (%) of full-length HIV-1 IN (FL) and truncation variants. Whereas NTDCCD indicate two-domain constructs, CCD/NTD indicate that single domain protein CCD and single domain NTD were purified separately and mixed together in the reaction. Values are means of three repeats (±SEM). P-values (one-way ANOVA) are < 0.05 (*) or < 0.01 (**) as calculated in comparison with FL sample, or < 0.05 (*') as compared to CCD/NTD. (B) CTD (blue) interactions with CCD (green) or NTD linker (magenta) at site-2 (S2). Dashed lines indicate hydrogen bonds made by K266 (K) or R228 (R) or CTD and Q53 (Q) of NTD. F: finger loop. Interacting residues are shown with lines. (C) NTD (magenta) and CTD (orange) docking at site-1 (S1) of CCD (green). F: finger loop. Interacting residues are shown with lines. (D) Structure of CCD (magenta, PDB code 3NF8) in complex with CDQ molecule (magenta sticks) bound at site-2. Superimposed is the CCD (green) of HIV-1 intasome (PDB code: 5U1C). Viral DNA (V DNA ) of HIV intasome is shown in orange. Binding of CDQ molecule displaces α4 by ~ 5 Å (measured at G149). F: finger loop.\",\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'TABREF1': {'text': '', 'latex': None, 'type': 'table'}},\n",
       " 'back_matter': [{'text': 'This work has been supported by iNEXT (grant number 653706, funded by the Horizon 2020 programme of the European',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Acknowledgments:'}]}"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['paper_id', 'metadata', 'abstract', 'body_text', 'bib_entries', 'ref_entries', 'back_matter'])"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "obj = data['body_text']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1282"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(obj[0]['text'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\"In the absence of a curative treatment, the highly active antiretroviral therapy (HAART) keeps the HIV-1 virus of AIDS patients under control. HAART combines drugs targeting different stages of viral replication including the integration step catalyzed by the integrase protein (IN) (1) . Integration of viral DNA into host genome involves two steps catalyzed by IN: (i) cleavage of a dinucleotide from each 3'-end of the viral DNA (3'processing), and (ii) insertion of this processed viral DNA into the host DNA (strand-transfer) (2) . Clinical IN strand transfer inhibitors (INSTIs) target the catalytic site of the enzyme to specifically inhibit the DNA joining reaction, however, as with all anti-AIDS treatments, the continued success of these drugs is persistently disrupted by resistance mutations (1, 2) . Although 3'-processing can be carried out by monomeric IN (3) , the assembly of IN functional multimers is imperative for the strand-transfer activity (4) (5) (6) (7) (8) , and for virus particle maturation and production (reviewed in (9, 10) ). In the continued quest to identify and develop new drugs, allosteric inhibitors that bind sites outside the catalytic core and disrupt IN multimerization are emerging with potent therapeutic potential (11) (12) (13) (14) .\""
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "obj[0]['text']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'(4)'"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "obj[1]['text'][436:439]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'start': 283, 'end': 286, 'text': '(1)', 'ref_id': 'BIBREF0'},\n",
       " {'start': 531, 'end': 534, 'text': '(2)', 'ref_id': 'BIBREF1'},\n",
       " {'start': 805, 'end': 808, 'text': '(1,', 'ref_id': 'BIBREF0'},\n",
       " {'start': 809, 'end': 811, 'text': '2)', 'ref_id': 'BIBREF1'},\n",
       " {'start': 869, 'end': 875, 'text': 'IN (3)', 'ref_id': 'BIBREF2'},\n",
       " {'start': 965, 'end': 968, 'text': '(4)', 'ref_id': 'BIBREF3'},\n",
       " {'start': 969, 'end': 972, 'text': '(5)', 'ref_id': 'BIBREF4'},\n",
       " {'start': 973, 'end': 976, 'text': '(6)', 'ref_id': 'BIBREF5'},\n",
       " {'start': 977, 'end': 980, 'text': '(7)', 'ref_id': 'BIBREF6'},\n",
       " {'start': 981, 'end': 984, 'text': '(8)', 'ref_id': 'BIBREF7'},\n",
       " {'start': 1049, 'end': 1052, 'text': '(9,', 'ref_id': 'BIBREF8'},\n",
       " {'start': 1053, 'end': 1056, 'text': '10)', 'ref_id': 'BIBREF9'},\n",
       " {'start': 1261, 'end': 1265, 'text': '(11)', 'ref_id': 'BIBREF10'},\n",
       " {'start': 1266, 'end': 1270, 'text': '(12)', 'ref_id': 'BIBREF11'},\n",
       " {'start': 1271, 'end': 1275, 'text': '(13)', 'ref_id': 'BIBREF12'},\n",
       " {'start': 1276, 'end': 1280, 'text': '(14)', 'ref_id': 'BIBREF13'}]"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "obj[0]['cite_spans']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'text': 'Hindering the assembly of IN functional multimers is only one side of the coin. Allosteric interference has also been shown to promote the formation of aberrant IN multimers and aggregates. The potential of allosteric IN inhibitors has been demonstrated through the thorough characterization of the \"LEDGF pocket\" formed at the dimer interface of IN and the development of LEDGIN (or ALLINI) inhibitors that bind to it (11) (Figure 1A) . Although less investigated, other IN pockets capable of allosteric inhibitor binding have also been identified ( Figure 1A) : binding of the \"Y3\" molecule to a pocket near the N-terminal end of CCD α-helix 4, designated Y3pocket, has been shown to inhibit 3\\'-processing and strand-transfer activities (16) ; the \"sucrose\" binding pocket found along the CCD dimer interface and flanked by two LEDGF pockets (17, 18) has recently been targeted by the natural product kuwanon-L, which inhibited IN activity in a pattern similar to LEDGINs (19) . Another fragment-binding pocket \"FBP\" has also been identified at the CCD dimer interface (20) . Although inhibitory profiles for this FBP pocket have not been shown, our previous work on IN from the feline virus showed that a single Phe187Lys mutation (Phe185 in HIV-1 IN) around this site inhibits dimer formation (3) . Fragment-based screening and structural studies have revealed the capability of all three pockets (Y3, FBP and LEDGF) to bind numerous small molecules (21) . Most recently, a new class of LEDGINs has been shown to distinctively bind to undefined interfaces of both CCD and CTD of IN (22) . Therefore, IN harbors several allosteric hotspots that can be explored for novel anti-multimerization intervention strategies.',\n",
       " 'cite_spans': [{'start': 419,\n",
       "   'end': 423,\n",
       "   'text': '(11)',\n",
       "   'ref_id': 'BIBREF10'},\n",
       "  {'start': 739, 'end': 743, 'text': '(16)', 'ref_id': 'BIBREF15'},\n",
       "  {'start': 844, 'end': 848, 'text': '(17,', 'ref_id': 'BIBREF16'},\n",
       "  {'start': 849, 'end': 852, 'text': '18)', 'ref_id': 'BIBREF17'},\n",
       "  {'start': 974, 'end': 978, 'text': '(19)', 'ref_id': 'BIBREF18'},\n",
       "  {'start': 1071, 'end': 1075, 'text': '(20)', 'ref_id': 'BIBREF19'},\n",
       "  {'start': 1297, 'end': 1300, 'text': '(3)', 'ref_id': 'BIBREF2'},\n",
       "  {'start': 1454, 'end': 1458, 'text': '(21)', 'ref_id': 'BIBREF20'},\n",
       "  {'start': 1586, 'end': 1590, 'text': '(22)', 'ref_id': 'BIBREF21'}],\n",
       " 'ref_spans': [{'start': 424,\n",
       "   'end': 435,\n",
       "   'text': '(Figure 1A)',\n",
       "   'ref_id': None},\n",
       "  {'start': 551, 'end': 561, 'text': 'Figure 1A)', 'ref_id': None}],\n",
       " 'section': ''}"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data['body_text'][2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[{'start': 419, 'end': 423, 'text': '(11)', 'ref_id': 'BIBREF10'}, {'start': 424, 'end': 435, 'text': '(Figure 1A)', 'ref_id': None}, {'start': 551, 'end': 561, 'text': 'Figure 1A)', 'ref_id': None}, {'start': 739, 'end': 743, 'text': '(16)', 'ref_id': 'BIBREF15'}, {'start': 844, 'end': 848, 'text': '(17,', 'ref_id': 'BIBREF16'}, {'start': 849, 'end': 852, 'text': '18)', 'ref_id': 'BIBREF17'}, {'start': 974, 'end': 978, 'text': '(19)', 'ref_id': 'BIBREF18'}, {'start': 1071, 'end': 1075, 'text': '(20)', 'ref_id': 'BIBREF19'}, {'start': 1297, 'end': 1300, 'text': '(3)', 'ref_id': 'BIBREF2'}, {'start': 1454, 'end': 1458, 'text': '(21)', 'ref_id': 'BIBREF20'}, {'start': 1586, 'end': 1590, 'text': '(22)', 'ref_id': 'BIBREF21'}]\n",
      "len text: 1719\n",
      "[(0, 419), (435, 551), (561, 739), (743, 844), (852, 974), (978, 1071), (1075, 1297), (1300, 1454), (1458, 1586), (1590, 1719)]\n",
      "(0, 419)\n",
      "Hindering the assembly of IN functional multimers is only one side of the coin. Allosteric interference has also been shown to promote the formation of aberrant IN multimers and aggregates. The potential of allosteric IN inhibitors has been demonstrated through the thorough characterization of the \"LEDGF pocket\" formed at the dimer interface of IN and the development of LEDGIN (or ALLINI) inhibitors that bind to it \n",
      "(435, 551)\n",
      " . Although less investigated, other IN pockets capable of allosteric inhibitor binding have also been identified ( \n",
      "(561, 739)\n",
      " : binding of the \"Y3\" molecule to a pocket near the N-terminal end of CCD α-helix 4, designated Y3pocket, has been shown to inhibit 3'-processing and strand-transfer activities \n",
      "(743, 844)\n",
      " ; the \"sucrose\" binding pocket found along the CCD dimer interface and flanked by two LEDGF pockets \n",
      "(852, 974)\n",
      " has recently been targeted by the natural product kuwanon-L, which inhibited IN activity in a pattern similar to LEDGINs \n",
      "(978, 1071)\n",
      " . Another fragment-binding pocket \"FBP\" has also been identified at the CCD dimer interface \n",
      "(1075, 1297)\n",
      " . Although inhibitory profiles for this FBP pocket have not been shown, our previous work on IN from the feline virus showed that a single Phe187Lys mutation (Phe185 in HIV-1 IN) around this site inhibits dimer formation \n",
      "(1300, 1454)\n",
      " . Fragment-based screening and structural studies have revealed the capability of all three pockets (Y3, FBP and LEDGF) to bind numerous small molecules \n",
      "(1458, 1586)\n",
      " . Most recently, a new class of LEDGINs has been shown to distinctively bind to undefined interfaces of both CCD and CTD of IN \n",
      "(1590, 1719)\n",
      " . Therefore, IN harbors several allosteric hotspots that can be explored for novel anti-multimerization intervention strategies.\n"
     ]
    }
   ],
   "source": [
    "obj = data['body_text'][2]\n",
    "span_keys = [\"cite_spans\", \"ref_spans\", \"eq_spans\"]\n",
    "all_spans = []\n",
    "for key in span_keys:\n",
    "    if key in obj.keys():\n",
    "        all_spans.extend(obj[key])\n",
    "#sort all_spans by start value\n",
    "all_spans = sorted(all_spans, key=lambda obj: obj['start'])\n",
    "print(all_spans)\n",
    "text = obj['text']\n",
    "print(f\"len text: {len(text)}\")\n",
    "keep_spans = []\n",
    "for i in range(len(all_spans)):\n",
    "    span = all_spans[i]\n",
    "    if i == 0:\n",
    "        keep_spans.append((0, span['start']))\n",
    "    else:\n",
    "        prev_span = all_spans[i-1]\n",
    "        if prev_span['end'] + 1 == span['start']:\n",
    "            continue\n",
    "        else:\n",
    "            keep_spans.append((prev_span['end'], span['start']))\n",
    "\n",
    "#add the end of the article\n",
    "keep_spans.append((all_spans[-1]['end'], len(text)))\n",
    "print(keep_spans)\n",
    "for span in keep_spans:\n",
    "    print(span)\n",
    "    print(text[span[0]:span[1]])\n",
    "#remove all the citations\n",
    "# new_text = \"\".join([text[span[0]:span[1]] for span in keep_spans])\n",
    "# #remove extra whitespace\n",
    "# new_text = re.sub(\"\\s+\", \" \", new_text)\n",
    "# print(new_text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'(17, 18)'"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "text[844:852]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "class JsonReader():\n",
    "    def __init__(self, file_path, remove_citations=True):\n",
    "        with open(file_path, \"r\") as f:\n",
    "            data = json.load(f)\n",
    "        self.extracted = defaultdict(str)\n",
    "        \n",
    "        text_keys = ['abstract', 'body_text'] #These require processing to clean citations\n",
    "        for key in text_keys:\n",
    "            if key in data.keys():\n",
    "                text_list = data[key]\n",
    "                cleaned_text = [obj['text'] for obj in text_list]    \n",
    "                if remove_citations:\n",
    "                    text_list = data[key]\n",
    "                    cleaned_text = list(map(self._remove_citations, text_list))\n",
    "            \n",
    "                #keep the texts according to their sections. Certain sections of papers may be more related\n",
    "                #than others such as the experiments section etc, helping to narrow down cluster formation\n",
    "                for i in range(len(text_list)):\n",
    "                    section_key = key+\".\"+text_list[i]['section']\n",
    "                    self.extracted[section_key] += cleaned_text[i]\n",
    "\n",
    "        self.extracted[\"paper_id\"] = data[\"paper_id\"]\n",
    "        if \"title\" in data.keys():\n",
    "            self.extracted[\"title\"] = data[\"title\"]  \n",
    "\n",
    "        \n",
    "    def _remove_citations(self, obj: dict):\n",
    "        \"\"\"\n",
    "        Text cleaning function to remove the citation spans from the text. \n",
    "        Args:\n",
    "            obj: dict, A dictionary of format\n",
    "                {\n",
    "                \"text\": <str>,\n",
    "                \"cite_spans\": [             # list of character indices of inline citations\n",
    "                                            # e.g. citation \"[7]\" occurs at positions 151-154 in \"text\"\n",
    "                                            #      linked to bibliography entry BIBREF3\n",
    "                    {\n",
    "                        \"start\": 151,\n",
    "                        \"end\": 154,\n",
    "                        \"text\": \"[7]\",\n",
    "                        \"ref_id\": \"BIBREF3\"\n",
    "                    },\n",
    "                    ...\n",
    "                ],\n",
    "                \"ref_spans\": <list of dicts similar to cite_spans>,     # e.g. inline reference to \"Table 1\"\n",
    "                \"section\": \"Abstract\"\n",
    "            }\n",
    "        \"\"\"\n",
    "        #Get a flat list of all the spans to remove\n",
    "        span_keys = [\"cite_spans\", \"ref_spans\", \"eq_spans\"]\n",
    "        all_spans = []\n",
    "        for key in span_keys:\n",
    "            if key in obj.keys():\n",
    "                all_spans.extend(obj[key])\n",
    "        \n",
    "        #sort all_spans by start value\n",
    "        all_spans = sorted(all_spans, key=lambda obj: obj['start'])\n",
    "        text = obj['text']\n",
    "        \n",
    "        if all_spans:\n",
    "            #Compute the complement: the list of indices that spans the text to be kept\n",
    "            keep_spans = []\n",
    "            for i in range(len(all_spans)):\n",
    "                span = all_spans[i]\n",
    "                if i == 0:\n",
    "                    keep_spans.append((0, span['start']))\n",
    "                else:\n",
    "                    prev_span = all_spans[i-1]\n",
    "                    if prev_span['end'] + 1 == span['start']:\n",
    "                        continue\n",
    "                    else:\n",
    "                        keep_spans.append((prev_span['end'], span['start']))\n",
    "            #add the end of the article to the spans\n",
    "            keep_spans.append((all_spans[-1]['end'], len(text)))\n",
    "\n",
    "            #remove all the citations\n",
    "            new_text = \"\".join([text[span[0]:span[1]] for span in keep_spans])\n",
    "            #remove extra whitespace\n",
    "            new_text = re.sub(\"\\s+\", \" \", new_text)\n",
    "            return new_text\n",
    "        \n",
    "        else:\n",
    "            return text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_extraction(json_file):\n",
    "    reader = JsonReader(json_file)\n",
    "    return reader.extracted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 5 µs, sys: 1 µs, total: 6 µs\n",
      "Wall time: 11.7 µs\n"
     ]
    }
   ],
   "source": [
    "%time\n",
    "\n",
    "extractions = list(map(get_extraction, all_jsons))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.DataFrame(extractions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>abstract.Abstract</th>\n",
       "      <th>body_text.</th>\n",
       "      <th>body_text.ACKNOWLEDGMENTS</th>\n",
       "      <th>paper_id</th>\n",
       "      <th>body_text.HIV/AIDS</th>\n",
       "      <th>body_text.Malaria</th>\n",
       "      <th>body_text.Tuberculosis</th>\n",
       "      <th>body_text.Influenza</th>\n",
       "      <th>body_text.West Nile Virus</th>\n",
       "      <th>body_text.SARS</th>\n",
       "      <th>...</th>\n",
       "      <th>body_text.Effect of U0126 on Initial Steps of Viral Replicative Cycle</th>\n",
       "      <th>body_text.Effect of U0126 on the Expression of JUNV Proteins</th>\n",
       "      <th>body_text.Effect of U0126 on Viral RNA Synthesis</th>\n",
       "      <th>body_text.Effect of U0126 on Viral mRNA Translation</th>\n",
       "      <th>body_text.Experimental Set-up</th>\n",
       "      <th>body_text.Preparation of DWV-Infected Honeybee Pupae</th>\n",
       "      <th>body_text.Feeding Regime</th>\n",
       "      <th>body_text.DWV Negative-Sense Strand Analyses</th>\n",
       "      <th>body_text.DWV Genomic Copies</th>\n",
       "      <th>body_text.Negative-Sense Strand Specific PCR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Bordetella bronchiseptica isolate KM22 has bee...</td>\n",
       "      <td>20-kb insert library preparation protocol (htt...</td>\n",
       "      <td>We thank the Yale University Center for Genome...</td>\n",
       "      <td>8f8eb4f004c2002face0723f2f58cc411954d36e</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>I nfectious diseases have been an ever-present...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>63f7049d200896290b38b38711113054f7ea1b50</td>\n",
       "      <td>HIV/AIDS has resulted in the death of &gt;20 mill...</td>\n",
       "      <td>The social, economic, and human toll exacted b...</td>\n",
       "      <td>Another ancient microbial scourge that has ree...</td>\n",
       "      <td>Each year, influenza develops in up to 20% of ...</td>\n",
       "      <td>West Nile virus (WNV), long endemic in Africa,...</td>\n",
       "      <td>The emergence of SARS in Asia in late 2002, an...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>The influenza A nucleoprotein (NP) is an attra...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4df45b8404d9de0b376a8ae3c282a517df36fe51</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>The outbreak of severe acute respiratory syndr...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>e0737ee93afe7b0bf06b1e3f9adf21d541dd10f0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>The prevalence of feline herpesvirus-1 (FHV-1)...</td>\n",
       "      <td>Feline herpesvirus type 1 (FHV-1) is the most ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3c3572ba243d61e7631725669c8f88347fdbd5bc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 90312 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   abstract.Abstract  \\\n",
       "0  Bordetella bronchiseptica isolate KM22 has bee...   \n",
       "1                                                NaN   \n",
       "2  The influenza A nucleoprotein (NP) is an attra...   \n",
       "3  The outbreak of severe acute respiratory syndr...   \n",
       "4  The prevalence of feline herpesvirus-1 (FHV-1)...   \n",
       "\n",
       "                                          body_text.  \\\n",
       "0  20-kb insert library preparation protocol (htt...   \n",
       "1  I nfectious diseases have been an ever-present...   \n",
       "2                                                NaN   \n",
       "3                                                NaN   \n",
       "4  Feline herpesvirus type 1 (FHV-1) is the most ...   \n",
       "\n",
       "                           body_text.ACKNOWLEDGMENTS  \\\n",
       "0  We thank the Yale University Center for Genome...   \n",
       "1                                                NaN   \n",
       "2                                                NaN   \n",
       "3                                                NaN   \n",
       "4                                                NaN   \n",
       "\n",
       "                                   paper_id  \\\n",
       "0  8f8eb4f004c2002face0723f2f58cc411954d36e   \n",
       "1  63f7049d200896290b38b38711113054f7ea1b50   \n",
       "2  4df45b8404d9de0b376a8ae3c282a517df36fe51   \n",
       "3  e0737ee93afe7b0bf06b1e3f9adf21d541dd10f0   \n",
       "4  3c3572ba243d61e7631725669c8f88347fdbd5bc   \n",
       "\n",
       "                                  body_text.HIV/AIDS  \\\n",
       "0                                                NaN   \n",
       "1  HIV/AIDS has resulted in the death of >20 mill...   \n",
       "2                                                NaN   \n",
       "3                                                NaN   \n",
       "4                                                NaN   \n",
       "\n",
       "                                   body_text.Malaria  \\\n",
       "0                                                NaN   \n",
       "1  The social, economic, and human toll exacted b...   \n",
       "2                                                NaN   \n",
       "3                                                NaN   \n",
       "4                                                NaN   \n",
       "\n",
       "                              body_text.Tuberculosis  \\\n",
       "0                                                NaN   \n",
       "1  Another ancient microbial scourge that has ree...   \n",
       "2                                                NaN   \n",
       "3                                                NaN   \n",
       "4                                                NaN   \n",
       "\n",
       "                                 body_text.Influenza  \\\n",
       "0                                                NaN   \n",
       "1  Each year, influenza develops in up to 20% of ...   \n",
       "2                                                NaN   \n",
       "3                                                NaN   \n",
       "4                                                NaN   \n",
       "\n",
       "                           body_text.West Nile Virus  \\\n",
       "0                                                NaN   \n",
       "1  West Nile virus (WNV), long endemic in Africa,...   \n",
       "2                                                NaN   \n",
       "3                                                NaN   \n",
       "4                                                NaN   \n",
       "\n",
       "                                      body_text.SARS  ...  \\\n",
       "0                                                NaN  ...   \n",
       "1  The emergence of SARS in Asia in late 2002, an...  ...   \n",
       "2                                                NaN  ...   \n",
       "3                                                NaN  ...   \n",
       "4                                                NaN  ...   \n",
       "\n",
       "  body_text.Effect of U0126 on Initial Steps of Viral Replicative Cycle  \\\n",
       "0                                                NaN                      \n",
       "1                                                NaN                      \n",
       "2                                                NaN                      \n",
       "3                                                NaN                      \n",
       "4                                                NaN                      \n",
       "\n",
       "  body_text.Effect of U0126 on the Expression of JUNV Proteins  \\\n",
       "0                                                NaN             \n",
       "1                                                NaN             \n",
       "2                                                NaN             \n",
       "3                                                NaN             \n",
       "4                                                NaN             \n",
       "\n",
       "  body_text.Effect of U0126 on Viral RNA Synthesis  \\\n",
       "0                                              NaN   \n",
       "1                                              NaN   \n",
       "2                                              NaN   \n",
       "3                                              NaN   \n",
       "4                                              NaN   \n",
       "\n",
       "  body_text.Effect of U0126 on Viral mRNA Translation  \\\n",
       "0                                                NaN    \n",
       "1                                                NaN    \n",
       "2                                                NaN    \n",
       "3                                                NaN    \n",
       "4                                                NaN    \n",
       "\n",
       "  body_text.Experimental Set-up  \\\n",
       "0                           NaN   \n",
       "1                           NaN   \n",
       "2                           NaN   \n",
       "3                           NaN   \n",
       "4                           NaN   \n",
       "\n",
       "  body_text.Preparation of DWV-Infected Honeybee Pupae  \\\n",
       "0                                                NaN     \n",
       "1                                                NaN     \n",
       "2                                                NaN     \n",
       "3                                                NaN     \n",
       "4                                                NaN     \n",
       "\n",
       "  body_text.Feeding Regime body_text.DWV Negative-Sense Strand Analyses  \\\n",
       "0                      NaN                                          NaN   \n",
       "1                      NaN                                          NaN   \n",
       "2                      NaN                                          NaN   \n",
       "3                      NaN                                          NaN   \n",
       "4                      NaN                                          NaN   \n",
       "\n",
       "  body_text.DWV Genomic Copies body_text.Negative-Sense Strand Specific PCR  \n",
       "0                          NaN                                          NaN  \n",
       "1                          NaN                                          NaN  \n",
       "2                          NaN                                          NaN  \n",
       "3                          NaN                                          NaN  \n",
       "4                          NaN                                          NaN  \n",
       "\n",
       "[5 rows x 90312 columns]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['abstract.Abstract', 'body_text.', 'body_text.ACKNOWLEDGMENTS',\n",
       "       'paper_id', 'body_text.HIV/AIDS', 'body_text.Malaria',\n",
       "       'body_text.Tuberculosis', 'body_text.Influenza',\n",
       "       'body_text.West Nile Virus', 'body_text.SARS',\n",
       "       ...\n",
       "       'body_text.Effect of U0126 on Initial Steps of Viral Replicative Cycle',\n",
       "       'body_text.Effect of U0126 on the Expression of JUNV Proteins',\n",
       "       'body_text.Effect of U0126 on Viral RNA Synthesis',\n",
       "       'body_text.Effect of U0126 on Viral mRNA Translation',\n",
       "       'body_text.Experimental Set-up',\n",
       "       'body_text.Preparation of DWV-Infected Honeybee Pupae',\n",
       "       'body_text.Feeding Regime',\n",
       "       'body_text.DWV Negative-Sense Strand Analyses',\n",
       "       'body_text.DWV Genomic Copies',\n",
       "       'body_text.Negative-Sense Strand Specific PCR'],\n",
       "      dtype='object', length=90312)"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "cord",
   "language": "python",
   "name": "cord"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
